---
title: "Strawberries Revamped"
author: "L Vilonen, CB Wall"
date: "5/5/2023"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 4
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '4'
---
## Project overview
  
Our project tested the influence of microbial communities on plant performance and how "home advantages" and environmental pressures alter plant-microbe interactions across a resource gradient (dry, intermediate, wet soils). We used a natural elevation and precipitation gradient in a factorial design experiment where we collected soil inoculum  across three sites of a watershed that increased in precipitation and elevation (henceforth called low, middle, and high elevation/precipitation sites).   

We used axenically cloned model plant system (strawberry, *Fragaria virginiana*) grown in sterile soil, inoculated with soil slurries from each of those sites, and subjected to reciprocal water treatments imposed at three levels of watering (dry, intermediate, wet) based on the average precipitation of the inoculum home environment.   

<center>  
  
![**Figure 1**. Inoculated strawberries in incubators](figures/Keck_strawberry_plants.jpg){width=50%}

</center>

We hypothesized that plants inoculated with soil microbes would outperform non-inoculated plants across all treatments and microbes from the low elevation site would be the most beneficial for plant growth under all water conditions. In testing the “home advantage” hypothesis, we expected plants from each inoculum treatment would perform best in their familiar water environment (e.g., low-inoculum plants would grow best under dry-water conditions, high-inoculum plants would grow best under wet-water conditions).

```{r setup chunk, include=FALSE, cache=FALSE, message=FALSE, warning=FALSE, collapse=TRUE}
knitr::opts_chunk$set(echo = TRUE)

# load packages
if (!require("pacman")) install.packages("pacman") # for rapid install if not in library

# use pacman to load all the packages you are missing!
pacman::p_load("car", "lme4", "lmerTest", "pbkrtest", "emmeans", "ggplot2", "tidyverse", "dplyr", "labdsv", "grid", "gridExtra", "patchwork", "plyr", "mctoolsr", "vegan", "car", "metacoder", "multcompView", "patchwork", "pairwiseAdonis", "labdsv", "eulerr", "ggthemes", "remotes", "mctoolsr", "brglm2", "cowplot", "microbiome", "mia")
```

### Plant Biomass
* ADD root/shoot graph codes here
* Make root/shoot weight graphs:  
* Read in the raw data and pair it down to what we're looking for.  

```{r plant biomass load, message=FALSE, warning=FALSE, results='hide'}
### Pair down raw data 
# raw data import
shootrootwithcontrol<-read.csv('data/strawberries_raw_full_data.csv',header=TRUE)
shootrootwithcontrol %>% drop_na()

#### # subset everything unnecessary

shootroot_step1removal<-subset(shootrootwithcontrol,TimePoint!="T0"&TimePoint!="Control"&Type!="Control")
shootroot_final<-shootroot_step1removal %>% drop_na()
shootroot_inoc23removed<-subset(shootroot_final,Inoculum!="2"&Inoculum!="3")
shootroot_inoc23removed$Inoculum<-as.factor(shootroot_inoc23removed$Inoculum)
shootroot_inoc23removed$Inoculum<-factor(shootroot_inoc23removed$Inoculum,levels=c("Dry Inoculation","Medium Inoculation","Wet Inoculation","No Inoculation"))

# rename levels
levels(shootroot_inoc23removed$Inoculum) <- list(
  Dry="Dry Inoculation", Moderate="Medium Inoculation", Wet="Wet Inoculation", Control="No Inoculation")

shootroot_inoc23removed$Water<-as.factor(shootroot_inoc23removed$Water)
shootroot_inoc23removed$Water<-factor(shootroot_inoc23removed$Water,levels=c("Low","Medium","High"))

# rename one level
levels(shootroot_inoc23removed$Water)[levels(shootroot_inoc23removed$Water)=="Medium"] <- "Moderate"

shootsonlywithcontrol<-subset(shootroot_inoc23removed,Type!="Root")
rootsonlywithcontrol<-subset(shootroot_inoc23removed,Type!="Shoot")

# export the final subset data
write.csv(shootsonlywithcontrol, "output/shootsonlywithcontrol.csv")
write.csv(rootsonlywithcontrol, "output/rootsonlywithcontrol.csv")
```

### Biomass proportion  
We take the root biomass (g) and the shoot biomass (g) to = total biomass, then express the proportion of the biomass for the roots and the shoots relative to total biomass of the plant. 

Stats for proportion above and belowground plant biomass
Run models linear models (Type III SS) for *proportion root or shoot biomass*  
* First we will run the linear model as `lm(mass ~ Water*Inoculation)`
* Then run post-hoc contrasts through `emmeans` (if significant effects observed)

```{r biomass prop, fig.show='hide'}

shootsonlywithcontrol <- shootsonlywithcontrol %>% 
       dplyr::rename("shoot.mass..g" = "WeightGrams")
rootsonlywithcontrol <- rootsonlywithcontrol %>% 
       dplyr::rename("root.mass..g" = "WeightGrams")
shootsonly_premerge<-subset(shootsonlywithcontrol,select = -c(SampleID,TimePoint,Type,SampleNotes,Water,Inoculum,Replicate,PlantNotes,SampleType,Sample_or_Control))
rootshoot_merge <- merge(shootsonly_premerge,rootsonlywithcontrol,by="PlantID")

#View(rootshoot_merge)
rootshoot_merge <-subset(rootshoot_merge, select = -c(Type,SampleNotes,PlantNotes))
rootshoot_merge$total.biomass..g<-rootshoot_merge$shoot.mass..g+rootshoot_merge$root.mass..g
rootshoot_merge$shoot.prop<-rootshoot_merge$shoot.mass..g/rootshoot_merge$total.biomass..g
rootshoot_merge$root.prop<-rootshoot_merge$root.mass..g/rootshoot_merge$total.biomass..g

#########
### From here with rootshoot_merge, you can redo all the above to see what the data looks like

hist(rootshoot_merge$root.prop)
hist(rootshoot_merge$shoot.prop)
```

Make a graph of the proportion of root and shoot biomass. *This is FIGURE 1.*
* For shoots : `shoot biomass (g) / total biomass (g)`
* For roots : `root biomass (g) / total biomass (g)` 

```{r biomass prop figure, width=20}

#View(rootshoot_merge)
longer_data <- rootshoot_merge %>%
  pivot_longer(shoot.prop:root.prop, names_to = "Type", values_to = "response")

longer_data$Type<-factor(longer_data$Type, levels=c("shoot.prop", "root.prop"))

## figure
prop.biomass<-ggplot(longer_data, aes(Water, response, fill=Inoculum)) +
  facet_wrap(.~Type)+
  geom_boxplot(outlier.size = 0.7, alpha=0.6)+
  ylab("Proportion of total biomass (g/g)") +
  #scale_fill_manual(values=c("orangered2", "goldenrod", "darkolivegreen3", "mistyrose2"))+
  scale_fill_brewer(palette="Dark2")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=18),
    panel.background = element_rect(fill=NA,color='black'),
    legend.key = element_rect(fill=NA),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))


prop.biomass
ggsave("figures/Fig1.prop.biomass.pdf", height=6, width=10)

```

Run models linear models (Type III SS) for proportion of total biomass as shoots and roots
* First we will run the linear model as `lm(prop mass ~ Water*Inoculation)`
* Then run post-hoc contrasts through `emmeans`
**Table 2A. Model and post-hocs for inoculation and water effects on proportion of *shoot* biomass**  
* Effect of water and inoculation
```{r prop shoots}
## This is table S1A
typeIIIModelshootprop<-lm(shoot.prop~Water*Inoculum,data=rootshoot_merge)
Anova(typeIIIModelshootprop,type=3)

#  posthoc
shoot.prop<-emmeans(typeIIIModelshootprop,pairwise~Inoculum|Water)
multcomp::cld(shoot.prop, Letters=letters)
```


**Table S2B. Model and post-hocs for inoculation and water effects on proportion of *root* biomass**  
* Effect of water*inoculation
```{r prop roots}
## This is table S1B
root.prop.mod<-lm(root.prop~Water*Inoculum,data=rootshoot_merge)
Anova(root.prop.mod,type=3)

# roots: Water posthoc
root.posthoc<-emmeans(root.prop.mod, ~Inoculum|Water)
multcomp::cld(root.posthoc, Letters=letters)
```

### Plant Survival
This is table S2, figure S2
```{r survival import and subset, results='hide',echo=FALSE}
library(dplyr) # may need to unload-reload if get an error here, plyr and dplyr don't play well

library(brglm2) # use this (bias-reduced logistic regression) due to complete separation
# Complete separation occurs in logistic (and binomial, and Poisson) regression when some categories contain 100% failures (or zero counts) or (in the logistic/binomial cases) 100% successes.
library(brglm2)

mortality.df<-read.csv("data/mortality_data.csv",header=TRUE)
mortality.df$mort.binary<- ifelse(mortality.df$mortality=="alive", 1, 0) # make binary
mortality.df<-subset(mortality.df,Inoculum!="2"&Inoculum!="3") # remove other inoculations
mortality.df$Inoculum=as.factor(mortality.df$Inoculum)
mortality.df$Water=as.factor(mortality.df$Water)

# full model
glm.mort.full<-glm(mort.binary ~ Inoculum*Water, family=binomial, data=mortality.df, method=brglmFit)
summary(glm.mort.full)
anova(glm.mort.full, test="Chisq") # p=0.009 for Inoculum

em.full.mort <- emmeans(glm.mort.full, pairwise~ Inoculum|Water)
multcomp::cld(em.full.mort, Letters=letters)

aggregate(mort.binary ~ Inoculum, data=mortality.df, FUN=sum)
```

Make a plot of survival data.
```{r survival prop data, results='hide'}
### final mortality data here
mortality_prop_data<-mortality.df %>%
  dplyr::group_by(Water, Inoculum) %>% 
  dplyr::summarize(prop = sum(mortality=="alive") / n())

mortality_prop_data$Water<-factor(mortality_prop_data$Water,
                                  levels=c("Low","Moderate","High"))

mortality_prop_data$Inoculum<-factor(mortality_prop_data$Inoculum,
                                        levels=c("Dry","Moderate","Wet","Control"))

### plot it
survival<-ggplot(mortality_prop_data,aes(Inoculum,prop,fill=Inoculum))+
  geom_bar(stat="identity", color="black", linewidth=0.3, alpha=0.7)+
  facet_wrap(.~Water)+
  ylab("Proportion Survival") +
  #scale_fill_manual(values=c("orangered2", "goldenrod", "darkolivegreen3", "mistyrose2"))+
  scale_fill_brewer(palette="Dark2")+
  xlab("Inoculum")+
  theme(
    axis.text.x=element_text(color='black',size=8),
    axis.title = element_text(color='black',size=10),
    axis.ticks = element_line(color='black'),
    legend.position = "none",
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

survival
ggsave("figures/FigS2.Survivorship.pdf", height=6, width=6)

#ggsave(filename = "mortality.pdf",
      #plot = mortality,
      #bg = "transparent",
      #width = 10, height = 6, units = "in",
      #dpi = 600)

```

### Fungi sequences
```{r, message=FALSE, warning=FALSE, results='hide'}
input <- load_taxa_table("data/ASV_table_postfiltering.txt", "data/map_loaded_strawberries.txt")
# 538 samples

#remove inoculum that we're not using, 350 samples remaining
input2 =filter_data(input, 'Inoculum', filter_vals = '2')
input3 =filter_data(input2, 'Inoculum', filter_vals = '3')

# final data with treatments to be used in analyses
ASV.data<-input3
ASV.data = filter_data(ASV.data, 'TimePoint', filter_vals = 'Control') # remove any controls
#346 samples remaining

otuTAB.all<-ASV.data$data_loaded
otuTAB.all <- as.data.frame(otuTAB.all)
otuTAB.transp.all<-t(otuTAB.all)
```

```{r phyloseq comparison}
# input.files, note that controls and slurry 2 and 3 would still be here
ASV.table<-read.delim("data/ASV_table_postfiltering.txt")
colnames(ASV.table)[colnames(ASV.table)=="X.OTU.ID"]<-"OTU.ID"

# adjust for rownames = OTU/ASVs
ASV.table <- ASV.table %>%
    tibble::column_to_rownames("OTU.ID") 

# remove taxonomy here for OTU table
ASV.table.edit<-within(ASV.table, rm(taxonomy))
OTU = otu_table(ASV.table.edit, taxa_are_rows = TRUE)

#### buildtaxonomy...
taxon<-as.data.frame(ASV.table$taxonomy)
rownames(taxon)<-rownames(ASV.table)
colnames(taxon)[colnames(taxon)=="ASV.table$taxonomy"]<-"taxonomy"

taxon.df<-tidyr::separate(taxon, taxonomy, into = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep = ";", fill = "right", extra = "drop")

# remove labels on names
taxon.df$Domain <- gsub('[k__]','',taxon.df$Domain)
taxon.df$Phylum <- gsub('[p__]','',taxon.df$Phylum)
taxon.df$Class <- gsub('[c__]','',taxon.df$Class)
taxon.df$Genus <- gsub('[g__]','',taxon.df$Genus)
taxon.df$Species <- gsub('[s__]','',taxon.df$Species)

taxon.mat<-as.matrix(taxon.df)
TAX<-tax_table(taxon.mat)

#### load in samples
samples_df<-read.delim("data/map_loaded_strawberries.txt")
colnames(samples_df)[colnames(samples_df)=="X.OTU.ID"]<-"OTU.ID"

samples_df$Water<- dplyr::recode_factor(samples_df$Water,
                "1" = "Low",
                "2" = "Moderate",
                "3" = "High",
                "NA"= "NA")

samples_df$Inoculum<- dplyr::recode_factor(samples_df$Inoculum,
                "1" = "Dry",
                "4" = "Moderate",
                "5" = "Wet",
                "6"= "Control")

samples_df$Trt<-interaction(samples_df$Water, samples_df$Inoculum)

# adjust for rownames = OTU/ASVs
samples_df <- samples_df %>%
    tibble::column_to_rownames("OTU.ID") 

# make phyloseq data
samples_data<-sample_data(samples_df)

### final objects
#TAX
#OTU
#samples_data

# phloseq merged
phy.straw<-phyloseq(OTU, TAX,samples_data)

#drop control samples
phy.straw.nocontrol = subset_samples(phy.straw, TimePoint != "Control") #534 samples

#drop inoc 2 and 3
phy.straw.drop.inoc = subset_samples(phy.straw.nocontrol, 
                                     Inoculum != "2" & Inoculum != "3") # 346 samples

#prune 0 counts
phy.straw.drop.inoc<-prune_taxa(taxa_sums(phy.straw.drop.inoc)> 0, phy.straw.drop.inoc)
# 346 samples, 1327 taxa/ASVs

write_phyloseq(phy.straw.drop.inoc, "output/phyloseq/")

#phy.straw.sh = subset_samples(phy.straw.drop.inoc, Type == "Shoot") #203 samples
#phy.straw.rt = subset_samples(phy.straw.drop.inoc, Type == "Root") #143 samples
```

```{r all sample rarefac, eval=FALSE}
# can be memory intensive, set to eval false for now
# make a rarefaction curve of all data

# set to eval =FALSE since can take some time to run
rarecurve(otuTAB.transp.all, xlab = "Sequencing Depth", col="mediumseagreen", ylab = "ASV count", label = FALSE, xlim=c(0,60000))

dev.copy(pdf, "figures/T1T0.all.depth.pdf", height=5, width=8)
dev.off() 
```

Then let's start with just looking at the basic NMDS plot between the
inoculations after the stress. Let's do a lot of filtering first.

```{r, message=FALSE, warning=FALSE, results='hide'}
# final data with treatments to be used in analyses 'ASV.data'

#keep only time point1, 284 samples
ASV.T0.sh = filter_data(ASV.data, 'TimePoint', keep_vals = 'T0') # 62 samples
ASV.T1 = filter_data(ASV.data, 'TimePoint', keep_vals = 'T1') # 284 samples
ASV.T1.rt = filter_data(ASV.T1, 'Type', keep_vals = 'Root') # 143 samples
ASV.T1.sh = filter_data(ASV.T1, 'Type', keep_vals = 'Shoot')# 203 samples

# all T1 data (post stress, roots and shoot)
T1.root.shoot<-ASV.T1

otuTAB.T1.root.shoot<-T1.root.shoot$data_loaded
otuTAB.T1.root.shoot <- as.data.frame (otuTAB.T1.root.shoot)
otuTAB.transp.t1.root.shoot<-t(otuTAB.T1.root.shoot)

```


```{r T1 rarefac, eval=FALSE}
# set to eval =FALSE since can take some time to run

# make a rarefaction curve of all T1 data
rarecurve(otuTAB.transp.t1.root.shoot, xlab = "Sequencing Depth", col="mediumseagreen", ylab = "ASV count", label = FALSE, xlim=c(0,60000))

dev.copy(pdf, "figures/T1.leaves_roots.depth.pdf", height=5, width=6)
dev.off() 

```

Hellinger Transformation only works after you've filtered the samples you
want (not sure why!). But be sure to filter first and then do the
transformation

```{r,results='hide', message=FALSE, warning=FALSE}
#using pack: labdsv

#Let's separate into roots and shoots for separate dataframes, valuable in analyses later on
roots_post_stress = filter_data(T1.root.shoot, 'Type', keep_vals = 'Root')
shoots_post_stress = filter_data(T1.root.shoot, 'Type', keep_vals = 'Shoot')

otuTAB.rootT1<-roots_post_stress$data_loaded
otuTAB.rootT1 <- as.data.frame (otuTAB.rootT1)
otuTABtranspose.root.T1<-t(otuTAB.rootT1)

hell.root.T1<-hellinger(otuTABtranspose.root.T1)
#View(hell)

roots_post_stress$data_loaded<-t(hell.root.T1)

### shoot T1 df
otuTAB.shootT1<-shoots_post_stress$data_loaded
otuTAB.shootT1 <- as.data.frame (otuTAB.shootT1)
otuTABtranspose.shoot.T1<-t(otuTAB.shootT1)
hell.shoot.T1<-hellinger(otuTABtranspose.shoot.T1)
#View(hell)

shoots_post_stress$data_loaded<-t(hell.shoot.T1)

```

Finally it is NMDS time yay! Permanova first

*Table 3* PERMANOVA testing water treatment and inoculum source effects on plant-associated fungal microbiomes in root and shoot biomass.
```{r shoot root NMDS}

########### SHOOTS
#you need to first load the data and make it a data frame. Then transpose the data frame. Add the map loaded too
otuTAB.sh.T1 <- as.data.frame(shoots_post_stress$data_loaded)
otuTAB.transp.sh.T1<-t(otuTAB.sh.T1)
mta.sh.T1<-shoots_post_stress$map_loaded

#do as factor
mta.sh.T1$Water <- as.factor(mta.sh.T1$Water)
mta.sh.T1$Inoculum <- as.factor(mta.sh.T1$Inoculum)

## using vegan
#make the model
set.seed(567)
model1.sh.T1<-adonis2(otuTAB.transp.sh.T1~Water*Inoculum,data=mta.sh.T1,permutations=999,method="bray")
model1.sh.T1 # interaction signif

mtapairwise.sh.T1<-unite(mta.sh.T1,c='water_inoculum',c('Water','Inoculum'),sep='-')

pair.mod.water.sh.T1<-pairwise.adonis(otuTAB.transp.sh.T1, factors=c(mtapairwise.sh.T1$water_inoculum))
pair.mod.water.sh.T1

# option if main effects were key
# pair.mod.sh.T1<-pairwise.adonis2(otuTAB.transp.sh.T1~Water/Inoculum, data=mta.sh.T1)


########### ROOTS
#you need to first load the data and make it a data frame. Then transpose the data frame. Add the map loaded too
otuTAB.ro.T1 <- as.data.frame (roots_post_stress$data_loaded)
otuTAB.transp.ro.T1<-t(otuTAB.ro.T1)
mta.ro.T1<-roots_post_stress$map_loaded

#do as factor
mta.ro.T1$Water <- as.factor(mta.ro.T1$Water)
mta.ro.T1$Inoculum <- as.factor(mta.ro.T1$Inoculum)

#make the model
set.seed(214)
model1.ro.T1<-adonis2(otuTAB.transp.ro.T1~Water*Inoculum,data=mta.ro.T1,permutations=999,method="bray")
model1.ro.T1

mtapairwise.ro.T1<-unite(mta.ro.T1,c='water_inoculum',c('Water','Inoculum'),sep='-')

# water pairwise
pair.mod.water.ro.T1<-pairwise.adonis(otuTAB.transp.ro.T1, factors=c(mta.ro.T1$Water))
pair.mod.water.ro.T1

# inoc pairwise
pair.mod.inoc.ro.T1<-pairwise.adonis(otuTAB.transp.ro.T1, factors=c(mta.ro.T1$Inoculum))
pair.mod.inoc.ro.T1

```

```{r roots only T1 for posthoc}
# separate models for each type (shoot and root) at T1, effect of water within inoculum.

####### separate by inoculum

################# Roots, T1
##### #inoc 1 
rt.T1.inoc1 = filter_data(ASV.T1.rt, 'Inoculum', keep_vals = '1') 
otuTAB.rt.T1.inoc1 <- as.data.frame (rt.T1.inoc1$data_loaded)
otuTAB.transp.rt.T1.inoc1<-t(otuTAB.rt.T1.inoc1)
mta.rt.T1.inoc1<-rt.T1.inoc1$map_loaded

#do as factor
mta.rt.T1.inoc1$Water <- as.factor(mta.rt.T1.inoc1$Water)
mta.rt.T1.inoc1$Inoculum <- as.factor(mta.rt.T1.inoc1$Inoculum)


##### #inoc 4
rt.T1.inoc4 = filter_data(ASV.T1.rt, 'Inoculum', keep_vals = '4') 
otuTAB.rt.T1.inoc4 <- as.data.frame (rt.T1.inoc4$data_loaded)
otuTAB.transp.rt.T1.inoc4<-t(otuTAB.rt.T1.inoc4)
mta.rt.T1.inoc4<-rt.T1.inoc4$map_loaded

#do as factor
mta.rt.T1.inoc4$Water <- as.factor(mta.rt.T1.inoc4$Water)
mta.rt.T1.inoc4$Inoculum <- as.factor(mta.rt.T1.inoc4$Inoculum)


###### #inoc 5
rt.T1.inoc5 = filter_data(ASV.T1.rt, 'Inoculum', keep_vals = '5') 
otuTAB.rt.T1.inoc5 <- as.data.frame (rt.T1.inoc5$data_loaded)
otuTAB.transp.rt.T1.inoc5<-t(otuTAB.rt.T1.inoc5)
mta.rt.T1.inoc5<-rt.T1.inoc5$map_loaded

#do as factor
mta.rt.T1.inoc5$Water <- as.factor(mta.rt.T1.inoc5$Water)
mta.rt.T1.inoc5$Inoculum <- as.factor(mta.rt.T1.inoc5$Inoculum)


######## #inoc 6
rt.T1.inoc6 = filter_data(ASV.T1.rt, 'Inoculum', keep_vals = '6') 
otuTAB.rt.T1.inoc6 <- as.data.frame (rt.T1.inoc6$data_loaded)
otuTAB.transp.rt.T1.inoc6<-t(otuTAB.rt.T1.inoc6)
mta.rt.T1.inoc6<-rt.T1.inoc6$map_loaded

#do as factor
mta.rt.T1.inoc6$Water <- as.factor(mta.rt.T1.inoc6$Water)
mta.rt.T1.inoc6$Inoculum <- as.factor(mta.rt.T1.inoc6$Inoculum)


####### Permanova, roots, separated by inoculum, T1

######### Inoc 1
#make the model
set.seed(213)
model1.ro.T1.inoc1<-adonis2(otuTAB.transp.rt.T1.inoc1~Water,data=mta.rt.T1.inoc1,
                      permutations=999,method="bray")
model1.ro.T1.inoc1

# water pairwise
pair.mod.water.ro.T1.inoc1<-pairwise.adonis(otuTAB.transp.rt.T1.inoc1, factors=c(mta.rt.T1.inoc1$Water),
                                            p.adjust.m = "hochberg")
pair.mod.water.ro.T1.inoc1


######### Inoc 4
#make the model
set.seed(808)
model1.ro.T1.inoc4<-adonis2(otuTAB.transp.rt.T1.inoc4~Water,data=mta.rt.T1.inoc4,
                      permutations=999,method="bray")
model1.ro.T1.inoc4

# water pairwise
pair.mod.water.ro.T1.inoc4<-pairwise.adonis(otuTAB.transp.rt.T1.inoc4, factors=c(mta.rt.T1.inoc4$Water),
                                            p.adjust.m = "hochberg")
pair.mod.water.ro.T1.inoc4



######### Inoc 5
#make the model
set.seed(999)
model1.ro.T1.inoc5<-adonis2(otuTAB.transp.rt.T1.inoc5~Water,data=mta.rt.T1.inoc5,
                      permutations=999,method="bray")
model1.ro.T1.inoc5

# water pairwise
pair.mod.water.ro.T1.inoc5<-pairwise.adonis(otuTAB.transp.rt.T1.inoc5, factors=c(mta.rt.T1.inoc5$Water),
                                            p.adjust.m = "hochberg")
pair.mod.water.ro.T1.inoc5

######### Inoc 6
#make the model
set.seed(138)
model1.ro.T1.inoc6<-adonis2(otuTAB.transp.rt.T1.inoc6~Water,data=mta.rt.T1.inoc6,
                      permutations=999,method="bray")
model1.ro.T1.inoc6

# water pairwise
pair.mod.water.ro.T1.inoc6<-pairwise.adonis(otuTAB.transp.rt.T1.inoc6, factors=c(mta.rt.T1.inoc6$Water),
                                            p.adjust.m = "hochberg")
pair.mod.water.ro.T1.inoc6


```


Everything is significant including pairwise adonis.

Let's make the plot:

```{r, results='hide', message=FALSE, warning=FALSE}
shoot.mds.T1<-metaMDS(t(shoots_post_stress$data_loaded))
shoot.T1<-scores(shoot.mds.T1)
shoot.T1.NMDS.df <- as.data.frame(shoot.T1$sites) 
shoot.T1.NMDS.df$Water <- shoots_post_stress$map_loaded$Water
shoot.T1.NMDS.df$Inoculum <- shoots_post_stress$map_loaded$Inoculum


root.mds.T1<-metaMDS(t(roots_post_stress$data_loaded))
root.T1<-scores(root.mds.T1)
root.T1.NMDS.df <- as.data.frame(root.T1$sites) 
root.T1.NMDS.df$Water <- as.factor(roots_post_stress$map_loaded$Water)
root.T1.NMDS.df$Inoculum <- as.factor(roots_post_stress$map_loaded$Inoculum)

```

*Figure 2. Shoot and Root NMDS at T1*
```{r, fig.height=5, fig.width=10, message=FALSE, warning=FALSE}
# change levels...
shoot.T1.NMDS.df$Inoculum <- as.factor(shoot.T1.NMDS.df$Inoculum)
levels(shoot.T1.NMDS.df$Inoculum)<-list(Dry="1", Moderate="4", Wet="5", Control="6")

# change levels...
shoot.T1.NMDS.df$Water <- as.factor(shoot.T1.NMDS.df$Water)
levels(shoot.T1.NMDS.df$Water) <- list(Low="1", Moderate="2", High="3")


# change levels...
root.T1.NMDS.df$Inoculum <- as.factor(root.T1.NMDS.df$Inoculum)
levels(root.T1.NMDS.df$Inoculum)<-list(Dry="1", Moderate="4", Wet="5", Control="6")

# change levels...
root.T1.NMDS.df$Water <- as.factor(root.T1.NMDS.df$Water)
levels(root.T1.NMDS.df$Water) <- list(Low="1", Moderate="2", High="3")

# Shoot T1 NMDS plot
Shoots.T1.NMDS<-ggplot(data=shoot.T1.NMDS.df,
                  aes(x=NMDS1,y=NMDS2,shape=Water,color=Inoculum)) + 
  scale_color_brewer(palette="Dark2")+
  ggtitle("Shoots T1")+
  geom_point(data=shoot.T1.NMDS.df,
             aes(x=NMDS1,y=NMDS2,shape=Water,color=Inoculum), alpha=0.8) + 
  # add the point markers
  stat_ellipse(data=shoot.T1.NMDS.df,aes(x=NMDS1,y=NMDS2,lty=Water,color=Inoculum))+
  scale_linetype_manual(values=c(1,2,3)) +
   theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))


### root T1 NMDS
Roots.T1.NMDS<-ggplot(data=root.T1.NMDS.df,
                      aes(x=NMDS1,y=NMDS2,shape=Water,color=Inoculum)) + 
  scale_color_brewer(palette="Dark2")+
  stat_ellipse(data=root.T1.NMDS.df,aes(x=NMDS1,y=NMDS2,lty=Water,color=Inoculum))+
  scale_linetype_manual(values=c(1,2,3)) +
  ggtitle("Roots T1")+
  geom_point(data=root.T1.NMDS.df,
             aes(x=NMDS1,y=NMDS2,shape=factor(Water),color=factor(Inoculum)), alpha=0.8) + 
  # add the point markers
 theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    legend.key = element_rect(fill=NA),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

both_plots<-Shoots.T1.NMDS+Roots.T1.NMDS
both_plots

ggsave("figures/Fig2.Shoot.Root.T1.NMDS.pdf", height=6, width=12)

#ggsave(filename = "both_plots.pdf",
      #plot = both_plots,
      #bg = "transparent",
      #width = 12, height = 6, units = "in",
      #dpi = 600)

```

### Shoots T0-T1
Refilter time. Select only shoots, compare how they change from T0 to T1

```{r, results='hide'}
#Let's view the original input map first
#View(input$map_loaded)
#Start by removing all roots. We can only do a time one with shoots

# final data with treatments to be used in analyses 'ASV.data'

shoot.T0T1 = filter_data(ASV.data, 'Type', keep_vals = 'Shoot') 
```

Then, let's set up the data for the NMDS plot:

```{r, results='hide', message=FALSE, warning=FALSE}
#View(shoot.T0T1$map_loaded)
shoot.T0T1.mds<-metaMDS(t(shoot.T0T1$data_loaded))
shoot.T0T1.NMDS.df <- as.data.frame(scores(shoot.T0T1.mds)$sites) 

shoot.T0T1.NMDS.df$Water <- as.factor(shoot.T0T1$map_loaded$Water)
shoot.T0T1.NMDS.df$Inoculum <- as.factor(shoot.T0T1$map_loaded$Inoculum)
shoot.T0T1.NMDS.df$TimePoint <- as.factor(shoot.T0T1$map_loaded$TimePoint)
shoot.T0T1.NMDS.df$PlantID <- shoot.T0T1$map_loaded$PlantID

#envfit
shoot.T0T1$map_loaded<-subset(shoot.T0T1$map_loaded, select = -c(WeightGrams,SampleNotes, PlantNotes))
shoot.T0T1$map_loaded$Water<-as.factor(shoot.T0T1$map_loaded$Water)
shoot.T0T1$map_loaded$Inoculum<-as.factor(shoot.T0T1$map_loaded$Inoculum)
shoot.T0T1$map_loaded$TimeWater <- paste(shoot.T0T1$map_loaded$TimePoint,shoot.T0T1$map_loaded$Water)
shoot.T0T1$map_loaded$TimeInoculum <- paste(shoot.T0T1$map_loaded$TimePoint,shoot.T0T1$map_loaded$Inoculum)

waterfinalvf <- envfit(shoot.T0T1.mds~TimeWater, data=shoot.T0T1$map_loaded, perm = 999)
water.scrs.final <- as.data.frame(scores(waterfinalvf, display = "factors"))

inocfinalvf <- envfit(shoot.T0T1.mds~TimeInoculum, data=shoot.T0T1$map_loaded, perm = 999)
inoc.scrs.final <- as.data.frame(scores(inocfinalvf, display = "factors"))


# change levels...
shoot.T0T1.NMDS.df$Inoculum <- as.factor(shoot.T0T1.NMDS.df$Inoculum)
levels(shoot.T0T1.NMDS.df$Inoculum)<-list(Dry="1", Moderate="4", Wet="5", Control="6")

# change levels...
shoot.T0T1.NMDS.df$Water <- as.factor(shoot.T0T1.NMDS.df$Water)
levels(shoot.T0T1.NMDS.df$Water) <- list(Low="1", Moderate="2", High="3")

```

```{r dropped figure, eval=FALSE}

#Time-by-inocultation NMDS. This figure may be cut.

NMDS1.mean.shoot.T0T1.df<-aggregate(NMDS1~ Water*TimePoint, data=shoot.T0T1.NMDS.df, FUN=mean)
NMDS2.mean.shoot.T0T1.df<-aggregate(NMDS2~ Water*TimePoint, data=shoot.T0T1.NMDS.df, FUN=mean)
centroid.shoot.T0T1.df<-as.data.frame(cbind(NMDS1.mean.shoot.T0T1.df, NMDS2.mean.shoot.T0T1.df[3]))


timewater<-ggplot(data=shoot.T0T1.NMDS.df,
                  aes(x=NMDS1,y=NMDS2,color=Water,shape=TimePoint), alpha=0.8) + 
  ggtitle("Water Treatment")+
  scale_color_manual(values=c("paleturquoise3", "steelblue2", "skyblue4"))+
  geom_point(data=shoot.T0T1.NMDS.df,aes(x=NMDS1,y=NMDS2,color=Water,
                                         shape=TimePoint)) +
  geom_segment(aes(x = centroid.shoot.T0T1.df[1,3], xend = centroid.shoot.T0T1.df[4,3], 
                   y = centroid.shoot.T0T1.df[1,4], yend = centroid.shoot.T0T1.df[4,4]),
               arrow = arrow(length = unit(0.25, "cm")), colour = "paleturquoise3") +
  geom_segment(aes(x = centroid.shoot.T0T1.df[2,3], xend = centroid.shoot.T0T1.df[5,3], 
                   y = centroid.shoot.T0T1.df[2,4], yend = centroid.shoot.T0T1.df[5,4]),
               arrow = arrow(length = unit(0.25, "cm")), colour = "steelblue2") +
  geom_segment(aes(x = centroid.shoot.T0T1.df[3,3], xend = centroid.shoot.T0T1.df[6,3], 
                   y = centroid.shoot.T0T1.df[3,4], yend = centroid.shoot.T0T1.df[6,4]),
               arrow = arrow(length = unit(0.25, "cm")), colour = "skyblue4") +
 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))

####
NMDS1.mean.data.inoc<-aggregate(NMDS1~ Inoculum*TimePoint, data=shoot.T0T1.NMDS.df, FUN=mean)
NMDS2.mean.data.inoc<-aggregate(NMDS2~ Inoculum*TimePoint, data=shoot.T0T1.NMDS.df, FUN=mean)
centroid.data.inoc<-as.data.frame(cbind(NMDS1.mean.data.inoc, NMDS2.mean.data.inoc[3]))


timeinoculation<-ggplot(data=shoot.T0T1.NMDS.df,
                        aes(x=NMDS1,y=NMDS2,color=Inoculum,shape=TimePoint)) + 
  scale_color_brewer(palette="Dark2")+ 
  ggtitle("Inoculum")+
  geom_point(data=shoot.T0T1.NMDS.df,aes(x=NMDS1,y=NMDS2,color=Inoculum,shape=TimePoint), alpha=0.8) + # add the point markers
  geom_segment(aes(x = centroid.data.inoc[1,3], xend = centroid.data.inoc[5,3], 
                   y = centroid.data.inoc[1,4], yend = centroid.data.inoc[5,4])
               ,arrow = arrow(length = unit(0.25, "cm")), colour = "#1B9E77") +
  geom_segment(aes(x = centroid.data.inoc[2,3], xend = centroid.data.inoc[6,3], 
                   y = centroid.data.inoc[2,4], yend = centroid.data.inoc[6,4]),
               arrow = arrow(length = unit(0.25, "cm")), colour = "#D95F02") +
  geom_segment(aes(x = centroid.data.inoc[3,3], xend = centroid.data.inoc[7,3], 
                   y = centroid.data.inoc[3,4], yend = centroid.data.inoc[7,4]),
               arrow = arrow(length = unit(0.25, "cm")), colour = "#7570B3") +
  geom_segment(aes(x = centroid.data.inoc[4,3], xend = centroid.data.inoc[8,3], 
                   y = centroid.data.inoc[4,4], yend = centroid.data.inoc[8,4]),
               arrow = arrow(length = unit(0.25, "cm")), colour = "#E7298A") +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    legend.key = element_rect(fill=NA),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))

# use packages patchwork
both_plots_move<-timewater+timeinoculation
both_plots_move

ggsave("figures/Figx.timeinoculation.pdf", height=6, width=8)


#ggsave(filename = "timeinoculation.pdf",
      #plot = timeinoculation,
      #bg = "transparent",
      #width = 8, height = 6, units = "in",
      #dpi = 600)
#geom_path(aes(color=Trt), show.legend = F)
```

Let's see if these are significant:
- this is *Table x and xx*
```{r T0T1 shoots Permanova and posthoc}
#shoots first: T0 to T1
#you need to first load the data and make it a data frame. Then transpose the data frame. Add the map loaded too
otuTAB.sh.T0T1 <- as.data.frame(shoot.T0T1$data_loaded)
otuTAB.transp.sh.T0T1<-t(otuTAB.sh.T0T1)
mta.sh.T0T1<-shoot.T0T1$map_loaded

#do as factor
mta.sh.T0T1$Water <- as.factor(mta.sh.T0T1$Water)
mta.sh.T0T1$Inoculum <- as.factor(mta.sh.T0T1$Inoculum)
mta.sh.T0T1$TimePoint <- as.factor(mta.sh.T0T1$TimePoint)


####### separate by inoculum
inputshootsonly = filter_data(ASV.data, 'Type', keep_vals = 'Shoot') 

## #inoc 1
sh.T0T1.inoc1 = filter_data(inputshootsonly, 'Inoculum', keep_vals = '1') 
otuTAB.sh.T0T1.inoc1 <- as.data.frame (sh.T0T1.inoc1$data_loaded)
otuTAB.transp.sh.T0T1.inoc1<-t(otuTAB.sh.T0T1.inoc1)
mta.sh.T0T1.inoc1<-sh.T0T1.inoc1$map_loaded

mta.sh.T0T1.inoc1$Water.T0.T1<-as.factor(ifelse(
              mta.sh.T0T1.inoc1$TimePoint=="T1" & mta.sh.T0T1.inoc1$Water=="1", "Low",
                ifelse(mta.sh.T0T1.inoc1$TimePoint=="T1" & mta.sh.T0T1.inoc1$Water=="2", "Moderate",
                  ifelse(mta.sh.T0T1.inoc1$TimePoint=="T1" & mta.sh.T0T1.inoc1$Water=="3", "High",
                    ifelse(mta.sh.T0T1.inoc1$TimePoint=="T0" & mta.sh.T0T1.inoc1$Water=="1", "T0",
              ifelse(mta.sh.T0T1.inoc1$TimePoint=="T0" & mta.sh.T0T1.inoc1$Water=="2", "T0", "T0"))))))
#do as factor
make.fac<-c("Water", "Inoculum", "TimePoint", "Water.T0.T1")
mta.sh.T0T1.inoc1[make.fac]<-lapply(mta.sh.T0T1.inoc1[make.fac], factor)

mta.sh.T0T1.inoc1$Water.T0.T1 <- factor(mta.sh.T0T1.inoc1$Water.T0.T1, levels=c("T0", "Low", "Moderate", "High"))


### inoc 4
sh.T0T1.inoc4 = filter_data(inputshootsonly, 'Inoculum', keep_vals = '4')
otuTAB.sh.T0T1.inoc4 <- as.data.frame (sh.T0T1.inoc4$data_loaded)
otuTAB.transp.sh.T0T1.inoc4<-t(otuTAB.sh.T0T1.inoc4)
mta.sh.T0T1.inoc4<-sh.T0T1.inoc4$map_loaded

mta.sh.T0T1.inoc4$Water.T0.T1<-as.factor(ifelse(
              mta.sh.T0T1.inoc4$TimePoint=="T1" & mta.sh.T0T1.inoc4$Water=="1", "Low",
                ifelse(mta.sh.T0T1.inoc4$TimePoint=="T1" & mta.sh.T0T1.inoc4$Water=="2", "Moderate",
                  ifelse(mta.sh.T0T1.inoc4$TimePoint=="T1" & mta.sh.T0T1.inoc4$Water=="3", "High",
                    ifelse(mta.sh.T0T1.inoc4$TimePoint=="T0" & mta.sh.T0T1.inoc4$Water=="1", "T0",
              ifelse(mta.sh.T0T1.inoc4$TimePoint=="T0" & mta.sh.T0T1.inoc4$Water=="2", "T0", "T0"))))))

mta.sh.T0T1.inoc4[make.fac]<-lapply(mta.sh.T0T1.inoc4[make.fac], factor)

mta.sh.T0T1.inoc4$Water.T0.T1 <- factor(mta.sh.T0T1.inoc4$Water.T0.T1, levels=c("T0", "Low", "Moderate", "High"))


## #inoc 5
sh.T0T1.inoc5 = filter_data(inputshootsonly, 'Inoculum', keep_vals = '5')
otuTAB.sh.T0T1.inoc5 <- as.data.frame (sh.T0T1.inoc5$data_loaded)
otuTAB.transp.sh.T0T1.inoc5<-t(otuTAB.sh.T0T1.inoc5)
mta.sh.T0T1.inoc5<-sh.T0T1.inoc5$map_loaded

mta.sh.T0T1.inoc5$Water.T0.T1<-as.factor(ifelse(
              mta.sh.T0T1.inoc5$TimePoint=="T1" & mta.sh.T0T1.inoc5$Water=="1", "Low",
                ifelse(mta.sh.T0T1.inoc5$TimePoint=="T1" & mta.sh.T0T1.inoc5$Water=="2", "Moderate",
                  ifelse(mta.sh.T0T1.inoc5$TimePoint=="T1" & mta.sh.T0T1.inoc5$Water=="3", "High",
                    ifelse(mta.sh.T0T1.inoc5$TimePoint=="T0" & mta.sh.T0T1.inoc5$Water=="1", "T0",
              ifelse(mta.sh.T0T1.inoc5$TimePoint=="T0" & mta.sh.T0T1.inoc5$Water=="2", "T0", "T0"))))))

mta.sh.T0T1.inoc5[make.fac]<-lapply(mta.sh.T0T1.inoc5[make.fac], factor)

mta.sh.T0T1.inoc5$Water.T0.T1 <- factor(mta.sh.T0T1.inoc5$Water.T0.T1, levels=c("T0", "Low", "Moderate", "High"))

### inoc 6
sh.T0T1.inoc6 = filter_data(inputshootsonly, 'Inoculum', keep_vals = '6') 
otuTAB.sh.T0T1.inoc6 <- as.data.frame (sh.T0T1.inoc6$data_loaded)
otuTAB.transp.sh.T0T1.inoc6<-t(otuTAB.sh.T0T1.inoc6)
mta.sh.T0T1.inoc6<-sh.T0T1.inoc6$map_loaded

mta.sh.T0T1.inoc6$Water.T0.T1<-as.factor(ifelse(
              mta.sh.T0T1.inoc6$TimePoint=="T1" & mta.sh.T0T1.inoc6$Water=="1", "Low",
                ifelse(mta.sh.T0T1.inoc6$TimePoint=="T1" & mta.sh.T0T1.inoc6$Water=="2", "Moderate",
                  ifelse(mta.sh.T0T1.inoc6$TimePoint=="T1" & mta.sh.T0T1.inoc6$Water=="3", "High",
                    ifelse(mta.sh.T0T1.inoc6$TimePoint=="T0" & mta.sh.T0T1.inoc6$Water=="1", "T0",
              ifelse(mta.sh.T0T1.inoc6$TimePoint=="T0" & mta.sh.T0T1.inoc6$Water=="2", "T0", "T0"))))))

mta.sh.T0T1.inoc6[make.fac]<-lapply(mta.sh.T0T1.inoc6[make.fac], factor)

mta.sh.T0T1.inoc6$Water.T0.T1 <- factor(mta.sh.T0T1.inoc6$Water.T0.T1, levels=c("T0", "Low", "Moderate", "High"))


############ inoc 1
mod1.sh.T0T1.inoc1<-adonis2(otuTAB.transp.sh.T0T1.inoc1~Water.T0.T1,data=mta.sh.T0T1.inoc1,permutations=999,method="bray")
mod1.sh.T0T1.inoc1

#use pack pairwiseAdonis
set.seed(777)
pair.mod.sh.T1<-pairwise.adonis(otuTAB.transp.sh.T0T1.inoc1,factors=mta.sh.T0T1.inoc1$Water.T0.T1, 
                                p.adjust.m = "hochberg")
pair.mod.sh.T1


############ inoc 4
mod1.sh.T0T1.inoc4<-adonis2(otuTAB.transp.sh.T0T1.inoc4~Water.T0.T1,data=mta.sh.T0T1.inoc4,permutations=999,method="bray")
mod1.sh.T0T1.inoc4

#use pack pairwiseAdonis
set.seed(808)
pair.mod.sh.T4<-pairwise.adonis(otuTAB.transp.sh.T0T1.inoc4,factors=mta.sh.T0T1.inoc4$Water.T0.T1, 
                                p.adjust.m = "hochberg")
pair.mod.sh.T4


########### inoc 5
mod1.sh.T0T1.inoc5<-adonis2(otuTAB.transp.sh.T0T1.inoc5~Water.T0.T1,data=mta.sh.T0T1.inoc5,permutations=999,method="bray")
mod1.sh.T0T1.inoc5

#use pack pairwiseAdonis
set.seed(805)
pair.mod.sh.T5<-pairwise.adonis(otuTAB.transp.sh.T0T1.inoc5,factors=mta.sh.T0T1.inoc5$Water.T0.T1,
                                p.adjust.m = "hochberg")
pair.mod.sh.T5

### inoc 6
mod1.sh.T0T1.inoc6<-adonis2(otuTAB.transp.sh.T0T1.inoc6~Water.T0.T1,data=mta.sh.T0T1.inoc6,permutations=999,method="bray")
mod1.sh.T0T1.inoc6

#use pack pairwiseAdonis
set.seed(972)
pair.mod.sh.T6<-pairwise.adonis(otuTAB.transp.sh.T0T1.inoc6,factors=mta.sh.T0T1.inoc6$Water.T0.T1,
                                p.adjust.m = "hochberg")
pair.mod.sh.T6



```

Let's look at T0 only
```{r, results='hide',echo=FALSE}
#Let's view the original input map first
#View(input$map_loaded)

# final data with treatments to be used in analyses 'ASV.data'

#keep only time point0, 62 samples
ASV.T0 = filter_data(ASV.data, 'TimePoint', keep_vals = 'T0')

# all T0 data (post stress, roots and shoot)
shoot.T0<-ASV.T0

# make a rarefaction curve of all T1 data
otuTAB.T0.shoot <- as.data.frame (shoot.T0$data_loaded)
otuTAB.transp.T0.shoot<-t(otuTAB.T0.shoot)

```

```{r T0 rarefac, eval=FALSE}
# make a rarefaction curve of all T0 leaf data
rarecurve(otuTAB.transp.t0.shoot, xlab = "Sequencing Depth", col="mediumseagreen", ylab = "ASV count", label = FALSE, xlim=c(0,20000))

dev.copy(pdf, "figures/T0.leaves.depth.pdf", height=5, width=6)
dev.off() 
```

```{r, results='hide',echo=FALSE}
### transform
hell.T0<-hellinger(otuTAB.transp.T0.shoot)

#View(hell)
shoot.T0$data_loaded<-t(hell.T0)

shoot.T0.mds<-metaMDS(t(shoot.T0$data_loaded))
shoot.T0.NMDS.df <- as.data.frame(scores(shoot.T0.mds)$sites) 
shoot.T0.NMDS.df$Water <- as.factor(shoot.T0$map_loaded$Water)
shoot.T0.NMDS.df$Inoculum <- as.factor(shoot.T0$map_loaded$Inoculum)
```

#### T0 Graph Only

Shoot fungal microbiomes prior to the start of water stress
*Figure S2: shoots T0 NMDS*
```{r, message=FALSE, warning=FALSE}

# change levels...
shoot.T0.NMDS.df$Inoculum <- as.factor(shoot.T0.NMDS.df$Inoculum)
levels(shoot.T0.NMDS.df$Inoculum)<-list(Dry="1", Moderate="4", Wet="5", Control="6")

# change levels...
shoot.T0.NMDS.df$Water <- as.factor(shoot.T0.NMDS.df$Water)
levels(shoot.T0.NMDS.df$Water) <- list(Low="1", Moderate="2", High="3")

# make the NMDS plot

Shoots.T0.NMDS<-ggplot(data=shoot.T0.NMDS.df,aes(x=NMDS1,y=NMDS2,color=Inoculum)) + 
  scale_color_brewer(palette="Dark2")+
  geom_point(data=shoot.T0.NMDS.df,
             aes(x=NMDS1,y=NMDS2,color=Inoculum), alpha=0.8) + 
  stat_ellipse(data=shoot.T0.NMDS.df,aes(x=NMDS1,y=NMDS2,color=Inoculum))+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key = element_rect(fill=NA),
    plot.title=element_text(size=20,face="bold",vjust=2))

Shoots.T0.NMDS
ggsave("figures/FigS3.Shoots.T0.NMDS.pdf", height=6, width=8)

```

*Table S3. T0 shoot fungal microbiome*
```{r shoot T0 microbiome}
library(vegan)

#make the model
otuTAB <- as.data.frame (ASV.T0$data_loaded)
otuTABtranspose<-t(otuTAB)
mta.sh.T0<-ASV.T0$map_loaded

#do as factor, no water treatment at T0
mta.sh.T0$Inoculum <- as.factor(mta.sh.T0$Inoculum)

set.seed(138)
model.sh.T0<-adonis2(otuTABtranspose~Inoculum,data=mta.sh.T0,permutations=999,method="bray")
model.sh.T0 # no effects
```

re-filter, shoots only, T0 and T1
```{r, results='hide',echo=FALSE}
inputshootsonly = filter_data(ASV.data, 'Type', keep_vals = 'Shoot') 

inoc1NMDSmovement =filter_data(inputshootsonly, 'Inoculum', keep_vals = '1')
inoc4NMDSmovement =filter_data(inputshootsonly, 'Inoculum', keep_vals = '4')
inoc5NMDSmovement =filter_data(inputshootsonly, 'Inoculum', keep_vals = '5')
inoc6NMDSmovement =filter_data(inputshootsonly, 'Inoculum', keep_vals = '6')
```

make otu dataframes for each inoculum (shoots only)
```{r,results='hide',echo=FALSE}
otuTAB<-inoc1NMDSmovement$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
inoc1NMDSmovement$data_loaded<-t(hell)

otuTAB<-inoc4NMDSmovement$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
inoc4NMDSmovement$data_loaded<-t(hell)

otuTAB<-inoc5NMDSmovement$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
inoc5NMDSmovement$data_loaded<-t(hell)

otuTAB<-inoc6NMDSmovement$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
inoc6NMDSmovement$data_loaded<-t(hell)
```

NMDS for each inoculum, shoots only
```{r, results='hide',echo=FALSE}
inoc1.mds<-metaMDS(t(inoc1NMDSmovement$data_loaded))
inoc1<-scores(inoc1.mds)
data.scoresinoc1 <- as.data.frame(inoc1$sites) 
data.scoresinoc1$Water <- inoc1NMDSmovement$map_loaded$Water
data.scoresinoc1$Inoculum <- inoc1NMDSmovement$map_loaded$Inoculum
data.scoresinoc1$TimePoint <- inoc1NMDSmovement$map_loaded$TimePoint
data.scoresinoc1$PlantID <- inoc1NMDSmovement$map_loaded$PlantID
inoc1NMDSmovement$map_loaded$TimeWater <- paste(inoc1NMDSmovement$map_loaded$TimePoint,inoc1NMDSmovement$map_loaded$Water)
inoc1vf <- envfit(inoc1.mds~TimeWater, data=inoc1NMDSmovement$map_loaded, perm = 999)
inoc1.scrs <- as.data.frame(scores(inoc1vf, display = "factors"))

inoc4.mds<-metaMDS(t(inoc4NMDSmovement$data_loaded))
inoc4<-scores(inoc4.mds)
data.scoresinoc4 <- as.data.frame(inoc4$sites) 
data.scoresinoc4$Water <- inoc4NMDSmovement$map_loaded$Water
data.scoresinoc4$Inoculum <- inoc4NMDSmovement$map_loaded$Inoculum
data.scoresinoc4$TimePoint <- inoc4NMDSmovement$map_loaded$TimePoint
data.scoresinoc4$PlantID <- inoc4NMDSmovement$map_loaded$PlantID
inoc4NMDSmovement$map_loaded$TimeWater <- paste(inoc4NMDSmovement$map_loaded$TimePoint,inoc4NMDSmovement$map_loaded$Water)
inoc4vf <- envfit(inoc4.mds~TimeWater, data=inoc4NMDSmovement$map_loaded, perm = 999)
inoc4.scrs <- as.data.frame(scores(inoc4vf, display = "factors"))

inoc5.mds<-metaMDS(t(inoc5NMDSmovement$data_loaded))
inoc5<-scores(inoc5.mds)
data.scoresinoc5 <- as.data.frame(inoc5$sites) 
data.scoresinoc5$Water <- inoc5NMDSmovement$map_loaded$Water
data.scoresinoc5$Inoculum <- inoc5NMDSmovement$map_loaded$Inoculum
data.scoresinoc5$TimePoint <- inoc5NMDSmovement$map_loaded$TimePoint
data.scoresinoc5$PlantID <- inoc5NMDSmovement$map_loaded$PlantID
inoc5NMDSmovement$map_loaded$TimeWater <- paste(inoc5NMDSmovement$map_loaded$TimePoint,inoc5NMDSmovement$map_loaded$Water)
inoc5vf <- envfit(inoc5.mds~TimeWater, data=inoc5NMDSmovement$map_loaded, perm = 999)
inoc5.scrs <- as.data.frame(scores(inoc5vf, display = "factors"))

inoc6.mds<-metaMDS(t(inoc6NMDSmovement$data_loaded))
inoc6<-scores(inoc6.mds)
data.scoresinoc6 <- as.data.frame(inoc6$sites) 
data.scoresinoc6$Water <- inoc6NMDSmovement$map_loaded$Water
data.scoresinoc6$Inoculum <- inoc6NMDSmovement$map_loaded$Inoculum
data.scoresinoc6$TimePoint <- inoc6NMDSmovement$map_loaded$TimePoint
data.scoresinoc6$PlantID <- inoc6NMDSmovement$map_loaded$PlantID
inoc6NMDSmovement$map_loaded$TimeWater <- paste(inoc6NMDSmovement$map_loaded$TimePoint,inoc6NMDSmovement$map_loaded$Water)
inoc6vf <- envfit(inoc6.mds~TimeWater, data=inoc6NMDSmovement$map_loaded, perm = 999)
inoc6.scrs <- as.data.frame(scores(inoc6vf, display = "factors"))
```

*Figure 3. NMDS for each inoculum (separate) by water treatment and vectors from T0 to T1*
```{r fig.height=15, fig.width=15, message=FALSE, warning=FALSE}

# combine all NMDS dataframes
all.NMDS.inoc<-as.data.frame(rbind(data.scoresinoc1, data.scoresinoc4, data.scoresinoc5, data.scoresinoc6))

# set structure
make.fac<-c("Water", "Inoculum", "TimePoint")
all.NMDS.inoc[make.fac] <- lapply(all.NMDS.inoc[make.fac], factor) # make all these factors
all.NMDS.inoc$TimePoint<-factor(all.NMDS.inoc$TimePoint, levels=c("T0", "T1"))


# change levels...
all.NMDS.inoc$Inoculum <- as.factor(all.NMDS.inoc$Inoculum)
levels(all.NMDS.inoc$Inoculum)<-list(Dry="1", Moderate="4", Wet="5", Control="6")

# change levels...
all.NMDS.inoc$Water <- as.factor(all.NMDS.inoc$Water)
levels(all.NMDS.inoc$Water) <- list(Low="1", Moderate="2", High="3")


# make new level
all.NMDS.inoc$Water.T0.T1<-as.factor(ifelse(
              all.NMDS.inoc$TimePoint=="T1" & all.NMDS.inoc$Water=="Low", "Low",
                ifelse(all.NMDS.inoc$TimePoint=="T1" & all.NMDS.inoc$Water=="Moderate", "Moderate",
                  ifelse(all.NMDS.inoc$TimePoint=="T1" & all.NMDS.inoc$Water=="High", "High",
                    ifelse(all.NMDS.inoc$TimePoint=="T0" & all.NMDS.inoc$Water=="Low", "T0",
              ifelse(all.NMDS.inoc$TimePoint=="T0" & all.NMDS.inoc$Water=="Moderate", "T0", "T0"))))))


all.NMDS.inoc$Water.T0.T1 <- factor(all.NMDS.inoc$Water.T0.T1, levels=c("T0", "Low", "Moderate", "High"))
                                       
####### Inoculum 1 centroids
NMDS1.mean.inoc1<-aggregate(NMDS1~ Water*TimePoint, data=data.scoresinoc1, FUN=mean)
NMDS2.mean.inoc1<-aggregate(NMDS2~ Water*TimePoint, data=data.scoresinoc1, FUN=mean)
centroid.inoc1<-as.data.frame(cbind(NMDS1.mean.inoc1, NMDS2.mean.inoc1[3]))
centroid.inoc1$TimePoint<-factor(centroid.inoc1$TimePoint, levels=c("T0", "T1"))
centroid.inoc1$Water<-as.factor(centroid.inoc1$Water)

# centroid for Water 0
NMDS1.W0.inoc1<-aggregate(NMDS1~ TimePoint, data=data.scoresinoc1, FUN=mean)
NMDS2.W0.inoc1<-aggregate(NMDS2~ TimePoint, data=data.scoresinoc1, FUN=mean)
centroid.W0.inoc1<-as.data.frame(cbind(NMDS1.W0.inoc1, NMDS2.W0.inoc1[2]))

# plot
inoc1.trajectory<-ggplot(subset(all.NMDS.inoc, Inoculum %in% "Dry"),
                         aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  ggtitle("Dry Inoculum")+
  geom_point(aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) +
  stat_ellipse(aes(x=NMDS1,y=NMDS2,color=factor(Water.T0.T1),linetype=factor(TimePoint)))+
  ylim(-2.2,2)+
  xlim(-2,2)+
  scale_color_manual(values=c("black", "paleturquoise3", "steelblue2", "skyblue4"))+
  scale_linetype_manual(values=c(2,1))+
  geom_segment(aes(x = centroid.W0.inoc1[1,2], xend = centroid.inoc1[4,3], 
                   y = centroid.W0.inoc1[1,3], yend = centroid.inoc1[4,4]), arrow = arrow(), colour = "paleturquoise3") +
  geom_segment(aes(x = centroid.W0.inoc1[1,2], xend = centroid.inoc1[5,3], 
                   y = centroid.W0.inoc1[1,3], yend = centroid.inoc1[5,4]),arrow = arrow(), colour = "steelblue2") +
  geom_segment(aes(x = centroid.W0.inoc1[1,2], xend = centroid.inoc1[6,3], 
                   y = centroid.W0.inoc1[1,3], yend = centroid.inoc1[6,4]),arrow = arrow(), colour = "skyblue4") +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key = element_rect(fill=NA),
    legend.position = "none",
    plot.title=element_text(size=16,face="bold",vjust=2))
###

####### Moderate 4
NMDS1.mean.inoc4<-aggregate(NMDS1~ Water*TimePoint, data=data.scoresinoc4, FUN=mean)
NMDS2.mean.inoc4<-aggregate(NMDS2~ Water*TimePoint, data=data.scoresinoc4, FUN=mean)
centroid.inoc4<-as.data.frame(cbind(NMDS1.mean.inoc4, NMDS2.mean.inoc4[3]))
centroid.inoc4$TimePoint<-factor(centroid.inoc4$TimePoint, levels=c("T0", "T1"))
centroid.inoc4$Water<-as.factor(centroid.inoc4$Water)

# centroid for Water 0
NMDS1.W0.inoc4<-aggregate(NMDS1~ TimePoint, data=data.scoresinoc4, FUN=mean)
NMDS2.W0.inoc4<-aggregate(NMDS2~ TimePoint, data=data.scoresinoc4, FUN=mean)
centroid.W0.inoc4<-as.data.frame(cbind(NMDS1.W0.inoc4, NMDS2.W0.inoc4[2]))

#plot
inoc4.trajectory<-ggplot(subset(all.NMDS.inoc, Inoculum %in% "Moderate"),
                         aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  ggtitle("Moderate Inoculum")+
  geom_point(aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  stat_ellipse(aes(x=NMDS1,y=NMDS2,color=factor(Water.T0.T1),linetype=factor(TimePoint)))+
  ylim(-2.2,2)+
  xlim(-2,2)+
  scale_color_manual(values=c("black", "paleturquoise3", "steelblue2", "skyblue4"))+ 
  scale_linetype_manual(values=c(2,1))+
  geom_segment(aes(x = centroid.W0.inoc4[1,2], xend = centroid.inoc4[4,3], 
                   y = centroid.W0.inoc4[1,3], yend = centroid.inoc4[4,4]),arrow = arrow(), colour = "paleturquoise3") +
  geom_segment(aes(x = centroid.W0.inoc4[1,2], xend = centroid.inoc4[5,3], 
                   y = centroid.W0.inoc4[1,3], yend = centroid.inoc4[5,4]),arrow = arrow(), colour = "steelblue2") +
  geom_segment(aes(x = centroid.W0.inoc4[1,2], xend = centroid.inoc4[6,3], 
                   y = centroid.W0.inoc4[1,3], yend = centroid.inoc4[6,4]),arrow = arrow(), colour = "skyblue4") +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    legend.key = element_rect(fill=NA),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
   
    plot.title=element_text(size=16,face="bold",vjust=2))

###

######### Inoculum 5

NMDS1.mean.inoc5<-aggregate(NMDS1~ Water*TimePoint, data=data.scoresinoc5, FUN=mean)
NMDS2.mean.inoc5<-aggregate(NMDS2~ Water*TimePoint, data=data.scoresinoc5, FUN=mean)
centroid.inoc5<-as.data.frame(cbind(NMDS1.mean.inoc5, NMDS2.mean.inoc5[3]))
centroid.inoc5$TimePoint<-factor(centroid.inoc5$TimePoint, levels=c("T0", "T1"))
centroid.inoc5$Water<-as.factor(centroid.inoc5$Water)

# centroid for Water 0
NMDS1.W0.inoc5<-aggregate(NMDS1~ TimePoint, data=data.scoresinoc5, FUN=mean)
NMDS2.W0.inoc5<-aggregate(NMDS2~ TimePoint, data=data.scoresinoc5, FUN=mean)
centroid.W0.inoc5<-as.data.frame(cbind(NMDS1.W0.inoc5, NMDS2.W0.inoc5[2]))

# plot
inoc5.trajectory<-ggplot(subset(all.NMDS.inoc, Inoculum %in% "Wet"),
                         aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  ggtitle("Wet Inoculum")+
  geom_point(aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  stat_ellipse(aes(x=NMDS1,y=NMDS2,color=factor(Water.T0.T1),linetype=factor(TimePoint)))+
  ylim(-2.2,2)+
  xlim(-2,2)+
  scale_color_manual(values=c("black", "paleturquoise3", "steelblue2", "skyblue4"))+ 
  scale_linetype_manual(values=c(2,1))+
  geom_segment(aes(x = centroid.W0.inoc5[1,2], xend = centroid.inoc5[4,3], 
                   y = centroid.W0.inoc5[1,3], yend = centroid.inoc5[4,4]),arrow = arrow(), colour = "paleturquoise3") +
  geom_segment(aes(x = centroid.W0.inoc5[1,2], xend = centroid.inoc5[5,3], 
                   y = centroid.W0.inoc5[1,3], yend = centroid.inoc5[5,4]),arrow = arrow(), colour = "steelblue2") +
  geom_segment(aes(x = centroid.W0.inoc5[1,2], xend = centroid.inoc5[6,3], 
                   y = centroid.W0.inoc5[1,3], yend = centroid.inoc5[6,4]),arrow = arrow(), colour = "skyblue4") +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    legend.key = element_rect(fill=NA),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=16,face="bold",vjust=2))

##

######### Inoculum 6

NMDS1.mean.inoc6<-aggregate(NMDS1~ Water*TimePoint, data=data.scoresinoc6, FUN=mean)
NMDS2.mean.inoc6<-aggregate(NMDS2~ Water*TimePoint, data=data.scoresinoc6, FUN=mean)
centroid.inoc6<-as.data.frame(cbind(NMDS1.mean.inoc6, NMDS2.mean.inoc6[3]))
centroid.inoc6$TimePoint<-factor(centroid.inoc6$TimePoint, levels=c("T0", "T1"))
centroid.inoc6$Water<-as.factor(centroid.inoc6$Water)

# centroid for Water 0
NMDS1.W0.inoc6<-aggregate(NMDS1~ TimePoint, data=data.scoresinoc6, FUN=mean)
NMDS2.W0.inoc6<-aggregate(NMDS2~ TimePoint, data=data.scoresinoc6, FUN=mean)
centroid.W0.inoc6<-as.data.frame(cbind(NMDS1.W0.inoc6, NMDS2.W0.inoc6[2]))

# plot
inoc6.trajectory<-ggplot(subset(all.NMDS.inoc, Inoculum %in% "Control"),
                         aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  ggtitle("Control")+
  geom_point(aes(x=NMDS1,y=NMDS2,shape=factor(TimePoint),color=factor(Water.T0.T1))) + 
  stat_ellipse(aes(x=NMDS1,y=NMDS2,color=factor(Water.T0.T1),linetype=factor(TimePoint)))+
  ylim(-2.2,2)+
  xlim(-2,2)+
  scale_color_manual(values=c("black", "paleturquoise3", "steelblue2", "skyblue4"))+ 
  scale_linetype_manual(values=c(2,1))+
  geom_segment(aes(x = centroid.W0.inoc6[1,2], xend = centroid.inoc6[4,3], 
                   y = centroid.W0.inoc6[1,3], yend = centroid.inoc6[4,4]),arrow = arrow(), colour = "paleturquoise3") +
  geom_segment(aes(x = centroid.W0.inoc6[1,2], xend = centroid.inoc6[5,3], 
                   y = centroid.W0.inoc6[1,3], yend = centroid.inoc6[5,4]),arrow = arrow(), colour = "steelblue2") +
  geom_segment(aes(x = centroid.W0.inoc6[1,2], xend = centroid.inoc6[6,3], 
                   y = centroid.W0.inoc6[1,3], yend = centroid.inoc6[6,4]),arrow = arrow(), colour = "skyblue4") +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    legend.key = element_rect(fill=NA),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=16,face="bold",vjust=2))


library(patchwork)
inocalone<-(inoc1.trajectory+inoc4.trajectory)/(inoc5.trajectory+inoc6.trajectory)
inocalone

inocalone
ggsave("figures/Fig3.T0T1.vector.pdf", height=8, width=8)
```


### Beta dispersion: Shoots T0 and T1 for 4 inoculums
```{r beta dispersion T0 to T1}
#### T0 and T1 shoots

# dry inoc
inoc1.dataframe <- as.data.frame(t(inoc1NMDSmovement$data_loaded))
groups <- inoc1NMDSmovement$map_loaded$TimePoint 
disinoc1 <- vegdist(inoc1.dataframe,method="bray")
bdinoc1<-betadisper(disinoc1,groups)
anova(bdinoc1)
pairwise.beta.disper.results.inoc1 <- permutest(bdinoc1, pairwise = TRUE, iter=9999)
pairwise.beta.disper.results.inoc1

# moderate inoc
inoc4.dataframe <- as.data.frame(t(inoc4NMDSmovement$data_loaded))
groups <- inoc4NMDSmovement$map_loaded$TimePoint
disinoc4 <- vegdist(inoc4.dataframe,method="bray")
bdinoc4<-betadisper(disinoc4,groups)
anova(bdinoc4)
pairwise.beta.disper.results.inoc4 <- permutest(bdinoc4, pairwise = TRUE, iter=9999)
pairwise.beta.disper.results.inoc4

# wet inoc
inoc5.dataframe <- as.data.frame(t(inoc5NMDSmovement$data_loaded))
groups <- inoc5NMDSmovement$map_loaded$TimePoint
disinoc5 <- vegdist(inoc5.dataframe,method="bray")
bdinoc5<-betadisper(disinoc5,groups)
anova(bdinoc5)
pairwise.beta.disper.results.inoc5 <- permutest(bdinoc5, pairwise = TRUE, iter=9999)
pairwise.beta.disper.results.inoc5

# control, uninoc
inoc6.dataframe <- as.data.frame(t(inoc6NMDSmovement$data_loaded))
groups <- inoc6NMDSmovement$map_loaded$TimePoint
disinoc6 <- vegdist(inoc6.dataframe,method="bray")
bdinoc6<-betadisper(disinoc6,groups)
anova(bdinoc6)
pairwise.beta.disper.results.inoc6 <- permutest(bdinoc6, pairwise = TRUE, iter=9999)
pairwise.beta.disper.results.inoc6
```

beta dispersion and mantel test (distance-difference in community by water). 
Let's do this with shoots and roots:
```{r, message=FALSE, warning=FALSE, results='hide',echo=FALSE, eval=FALSE}

library(labdsv)

#keep only time point1
inputtimepoint1only = filter_data(ASV.data, 'TimePoint', keep_vals = 'T1')
#Let's separate into roots and shoots
roots_post_stress = filter_data(inputstep3, 'Type', keep_vals = 'Root')
shoots_post_stress = filter_data(inputstep3, 'Type', keep_vals = 'Shoot')

otuTAB<-roots_post_stress$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roots_post_stress$data_loaded<-t(hell)

otuTAB<-shoots_post_stress$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoots_post_stress$data_loaded<-t(hell)

############
## mantel test: (significant across inoculum, but not water treatment)

roots.dataframe<-as.data.frame(t(roots_post_stress$data_loaded))
dist.abund.root = vegdist(roots.dataframe, method = "bray")
inoc <- roots_post_stress$map_loaded$Inoculum
water <- roots_post_stress$map_loaded$Water
dist.inoc = dist(inoc, method = "euclidean")
dist.water = dist(water, method = "euclidean")
abund_inoc = mantel(dist.abund.root, dist.inoc, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_inoc
abund_water = mantel(dist.abund.root, dist.water, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_water

##########
shoots.dataframe<-as.data.frame(t(shoots_post_stress$data_loaded))
dist.abund.shoot = vegdist(shoots.dataframe, method = "bray")
inoc <- shoots_post_stress$map_loaded$Inoculum
water <- shoots_post_stress$map_loaded$Water
dist.inoc = dist(inoc, method = "euclidean")
dist.water = dist(water, method = "euclidean")
abund_inoc = mantel(dist.abund.shoot, dist.inoc, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_inoc
abund_water = mantel(dist.abund.shoot, dist.water, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_water


```

### Shannon's diversity
Let's make the shannon's diversity plots next: 
- for T1 roots
- for T0 and T1 shoots

```{r, shannon diversity, results='hide'}
# inputraw has the slurry data so let's take that first. Then we will need to transform it too. 
# filtered data is 'ASV.data'

inputstep5 = filter_data(ASV.data, 'TimePoint', keep_vals = 'T1')
roott1 = filter_data(inputstep5, 'Type', keep_vals = 'Root')

# T1: Roots, water 1
roott1water1 = filter_data(roott1, 'Water', keep_vals = '1')
roott1inoc1water1 = filter_data(roott1water1, 'Inoculum', keep_vals = '1')
roott1inoc4water1 = filter_data(roott1water1, 'Inoculum', keep_vals = '4')
roott1inoc5water1 = filter_data(roott1water1, 'Inoculum', keep_vals = '5')
roott1inoc6water1 = filter_data(roott1water1, 'Inoculum', keep_vals = '6')

# T1: Roots, water 2
roott1water2 = filter_data(roott1, 'Water', keep_vals = '2')
roott1inoc1water2 = filter_data(roott1water2, 'Inoculum', keep_vals = '1')
roott1inoc4water2 = filter_data(roott1water2, 'Inoculum', keep_vals = '4')
roott1inoc5water2 = filter_data(roott1water2, 'Inoculum', keep_vals = '5')
roott1inoc6water2 = filter_data(roott1water2, 'Inoculum', keep_vals = '6')

# T1: Roots, water 3
roott1water3 = filter_data(roott1, 'Water', keep_vals = '3')
roott1inoc1water3 = filter_data(roott1water3, 'Inoculum', keep_vals = '1')
roott1inoc4water3 = filter_data(roott1water3, 'Inoculum', keep_vals = '4')
roott1inoc5water3 = filter_data(roott1water3, 'Inoculum', keep_vals = '5')
roott1inoc6water3 = filter_data(roott1water3, 'Inoculum', keep_vals = '6')

# T1: Shoots, water 1
shoott1 = filter_data(inputstep5, 'Type', keep_vals = 'Shoot')
shoott1water1 = filter_data(shoott1, 'Water', keep_vals = '1')
shoott1inoc1water1 = filter_data(shoott1water1, 'Inoculum', keep_vals = '1')
shoott1inoc4water1 = filter_data(shoott1water1, 'Inoculum', keep_vals = '4')
shoott1inoc5water1 = filter_data(shoott1water1, 'Inoculum', keep_vals = '5')
shoott1inoc6water1 = filter_data(shoott1water1, 'Inoculum', keep_vals = '6')

# T1: Shoots, water 2
shoott1water2 = filter_data(shoott1, 'Water', keep_vals = '2')
shoott1inoc1water2 = filter_data(shoott1water2, 'Inoculum', keep_vals = '1')
shoott1inoc4water2 = filter_data(shoott1water2, 'Inoculum', keep_vals = '4')
shoott1inoc5water2 = filter_data(shoott1water2, 'Inoculum', keep_vals = '5')
shoott1inoc6water2 = filter_data(shoott1water2, 'Inoculum', keep_vals = '6')

# T1: Shoots, water 3
shoott1water3 = filter_data(shoott1, 'Water', keep_vals = '3')
shoott1inoc1water3 = filter_data(shoott1water3, 'Inoculum', keep_vals = '1')
shoott1inoc4water3 = filter_data(shoott1water3, 'Inoculum', keep_vals = '4')
shoott1inoc5water3 = filter_data(shoott1water3, 'Inoculum', keep_vals = '5')
shoott1inoc6water3 = filter_data(shoott1water3, 'Inoculum', keep_vals = '6')

# T0
t0 = filter_data(input, 'TimePoint', keep_vals = 'T0')
shoott0 = filter_data(t0, 'Type', keep_vals = 'Shoot')

# T0: Shoots, water T0
shoott0inoc1 = filter_data(shoott0, 'Inoculum', keep_vals = '1')
shoott0inoc4 = filter_data(shoott0, 'Inoculum', keep_vals = '4')
shoott0inoc5 = filter_data(shoott0, 'Inoculum', keep_vals = '5')
shoott0inoc6 = filter_data(shoott0, 'Inoculum', keep_vals = '6')

#Now that we have all three ready, let's transform them to match our other data:

otuTAB<-roott1inoc1water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc1water1$data_loaded<-t(hell)

otuTAB<-roott1inoc4water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc4water1$data_loaded<-t(hell)

otuTAB<-roott1inoc5water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc5water1$data_loaded<-t(hell)

otuTAB<-roott1inoc6water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc6water1$data_loaded<-t(hell)

otuTAB<-roott1inoc1water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc1water2$data_loaded<-t(hell)

otuTAB<-roott1inoc4water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc4water2$data_loaded<-t(hell)

otuTAB<-roott1inoc5water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc5water2$data_loaded<-t(hell)

otuTAB<-roott1inoc6water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc6water2$data_loaded<-t(hell)

otuTAB<-roott1inoc1water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc1water3$data_loaded<-t(hell)

otuTAB<-roott1inoc4water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc4water3$data_loaded<-t(hell)

otuTAB<-roott1inoc5water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc5water3$data_loaded<-t(hell)

otuTAB<-roott1inoc6water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
roott1inoc6water3$data_loaded<-t(hell)

otuTAB<-shoott1inoc1water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc1water1$data_loaded<-t(hell)

otuTAB<-shoott1inoc4water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc4water1$data_loaded<-t(hell)

otuTAB<-shoott1inoc5water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc5water1$data_loaded<-t(hell)

otuTAB<-shoott1inoc6water1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc6water1$data_loaded<-t(hell)

otuTAB<-shoott1inoc1water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc1water2$data_loaded<-t(hell)

otuTAB<-shoott1inoc4water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc4water2$data_loaded<-t(hell)

otuTAB<-shoott1inoc5water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc5water2$data_loaded<-t(hell)

otuTAB<-shoott1inoc6water2$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc6water2$data_loaded<-t(hell)

otuTAB<-shoott1inoc1water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc1water3$data_loaded<-t(hell)

otuTAB<-shoott1inoc4water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc4water3$data_loaded<-t(hell)

otuTAB<-shoott1inoc5water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc5water3$data_loaded<-t(hell)

otuTAB<-shoott1inoc6water3$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott1inoc6water3$data_loaded<-t(hell)

otuTAB<-shoott0inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott0inoc1$data_loaded<-t(hell)

otuTAB<-shoott0inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott0inoc4$data_loaded<-t(hell)

otuTAB<-shoott0inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott0inoc5$data_loaded<-t(hell)

otuTAB<-shoott0inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
shoott0inoc6$data_loaded<-t(hell)
```

Now that we have the data, we can do shannon's diversity:
- for each treatment roots at T1
- for each treatment shoots and T0 and T1
```{r,results='hide'}
#all shannon's diversity
library(vegan)
roott1inoc1water1<-diversity(t(roott1inoc1water1$data_loaded), index = "shannon")
roott1inoc4water1<-diversity(t(roott1inoc4water1$data_loaded), index = "shannon")
roott1inoc5water1<-diversity(t(roott1inoc5water1$data_loaded), index = "shannon")
roott1inoc6water1<-diversity(t(roott1inoc6water1$data_loaded), index = "shannon")

roott1inoc1water2<-diversity(t(roott1inoc1water2$data_loaded), index = "shannon")
roott1inoc4water2<-diversity(t(roott1inoc4water2$data_loaded), index = "shannon")
roott1inoc5water2<-diversity(t(roott1inoc5water2$data_loaded), index = "shannon")
roott1inoc6water2<-diversity(t(roott1inoc6water2$data_loaded), index = "shannon")

roott1inoc1water3<-diversity(t(roott1inoc1water3$data_loaded), index = "shannon")
roott1inoc4water3<-diversity(t(roott1inoc4water3$data_loaded), index = "shannon")
roott1inoc5water3<-diversity(t(roott1inoc5water3$data_loaded), index = "shannon")
roott1inoc6water3<-diversity(t(roott1inoc6water3$data_loaded), index = "shannon")

shoott1inoc1water1<-diversity(t(shoott1inoc1water1$data_loaded), index = "shannon")
shoott1inoc4water1<-diversity(t(shoott1inoc4water1$data_loaded), index = "shannon")
shoott1inoc5water1<-diversity(t(shoott1inoc5water1$data_loaded), index = "shannon")
shoott1inoc6water1<-diversity(t(shoott1inoc6water1$data_loaded), index = "shannon")

shoott1inoc1water2<-diversity(t(shoott1inoc1water2$data_loaded), index = "shannon")
shoott1inoc4water2<-diversity(t(shoott1inoc4water2$data_loaded), index = "shannon")
shoott1inoc5water2<-diversity(t(shoott1inoc5water2$data_loaded), index = "shannon")
shoott1inoc6water2<-diversity(t(shoott1inoc6water2$data_loaded), index = "shannon")

shoott1inoc1water3<-diversity(t(shoott1inoc1water3$data_loaded), index = "shannon")
shoott1inoc4water3<-diversity(t(shoott1inoc4water3$data_loaded), index = "shannon")
shoott1inoc5water3<-diversity(t(shoott1inoc5water3$data_loaded), index = "shannon")
shoott1inoc6water3<-diversity(t(shoott1inoc6water3$data_loaded), index = "shannon")

shoott0inoc1<-diversity(t(shoott0inoc1$data_loaded), index = "shannon")
shoott0inoc4<-diversity(t(shoott0inoc4$data_loaded), index = "shannon")
shoott0inoc5<-diversity(t(shoott0inoc5$data_loaded), index = "shannon")
shoott0inoc6<-diversity(t(shoott0inoc6$data_loaded), index = "shannon")
```

Next, lets make the dataframe for the ggplot - shannon's (violin plot). We need to make sure we can panel it out too.
```{r,results='hide'}
#roott1inoc1water1
roott1inoc1water1df<-as.data.frame(roott1inoc1water1)
colnames(roott1inoc1water1df)[1] ="shannon"
water<-rep("water1",times=nrow(roott1inoc1water1df))
inoculum<-rep("inoc1",times=nrow(roott1inoc1water1df))
type<-rep("root",times=nrow(roott1inoc1water1df))
roott1inoc1water1dffinal<-cbind(roott1inoc1water1df,water,inoculum,type)

#roott1inoc4water1
roott1inoc4water1df<-as.data.frame(roott1inoc4water1)
colnames(roott1inoc4water1df)[1] ="shannon"
water<-rep("water1",times=nrow(roott1inoc4water1df))
inoculum<-rep("inoc4",times=nrow(roott1inoc4water1df))
type<-rep("root",times=nrow(roott1inoc4water1df))
roott1inoc4water1dffinal<-cbind(roott1inoc4water1df,water,inoculum,type)

#roott1inoc5water1
roott1inoc5water1df<-as.data.frame(roott1inoc5water1)
colnames(roott1inoc5water1df)[1] ="shannon"
water<-rep("water1",times=nrow(roott1inoc5water1df))
inoculum<-rep("inoc5",times=nrow(roott1inoc5water1df))
type<-rep("root",times=nrow(roott1inoc5water1df))
roott1inoc5water1dffinal<-cbind(roott1inoc5water1df,water,inoculum,type)

#roott1inoc6water1
roott1inoc6water1df<-as.data.frame(roott1inoc6water1)
colnames(roott1inoc6water1df)[1] ="shannon"
water<-rep("water1",times=nrow(roott1inoc6water1df))
inoculum<-rep("inoc6",times=nrow(roott1inoc6water1df))
type<-rep("root",times=nrow(roott1inoc6water1df))
roott1inoc6water1dffinal<-cbind(roott1inoc6water1df,water,inoculum,type)

#roott1inoc1water2
roott1inoc1water2df<-as.data.frame(roott1inoc1water2)
colnames(roott1inoc1water2df)[1] ="shannon"
water<-rep("water2",times=nrow(roott1inoc1water2df))
inoculum<-rep("inoc1",times=nrow(roott1inoc1water2df))
type<-rep("root",times=nrow(roott1inoc1water2df))
roott1inoc1water2dffinal<-cbind(roott1inoc1water2df,water,inoculum,type)

#roott1inoc4water2
roott1inoc4water2df<-as.data.frame(roott1inoc4water2)
colnames(roott1inoc4water2df)[1] ="shannon"
water<-rep("water2",times=nrow(roott1inoc4water2df))
inoculum<-rep("inoc4",times=nrow(roott1inoc4water2df))
type<-rep("root",times=nrow(roott1inoc4water2df))
roott1inoc4water2dffinal<-cbind(roott1inoc4water2df,water,inoculum,type)

#roott1inoc5water2
roott1inoc5water2df<-as.data.frame(roott1inoc5water2)
colnames(roott1inoc5water2df)[1] ="shannon"
water<-rep("water2",times=nrow(roott1inoc5water2df))
inoculum<-rep("inoc5",times=nrow(roott1inoc5water2df))
type<-rep("root",times=nrow(roott1inoc5water2df))
roott1inoc5water2dffinal<-cbind(roott1inoc5water2df,water,inoculum,type)

#roott1inoc6water2
roott1inoc6water2df<-as.data.frame(roott1inoc6water2)
colnames(roott1inoc6water2df)[1] ="shannon"
water<-rep("water2",times=nrow(roott1inoc6water2df))
inoculum<-rep("inoc6",times=nrow(roott1inoc6water2df))
type<-rep("root",times=nrow(roott1inoc6water2df))
roott1inoc6water2dffinal<-cbind(roott1inoc6water2df,water,inoculum,type)

#roott1inoc1water3
roott1inoc1water3df<-as.data.frame(roott1inoc1water3)
colnames(roott1inoc1water3df)[1] ="shannon"
water<-rep("water3",times=nrow(roott1inoc1water3df))
inoculum<-rep("inoc1",times=nrow(roott1inoc1water3df))
type<-rep("root",times=nrow(roott1inoc1water3df))
roott1inoc1water3dffinal<-cbind(roott1inoc1water3df,water,inoculum,type)

#roott1inoc4water3
roott1inoc4water3df<-as.data.frame(roott1inoc4water3)
colnames(roott1inoc4water3df)[1] ="shannon"
water<-rep("water3",times=nrow(roott1inoc4water3df))
inoculum<-rep("inoc4",times=nrow(roott1inoc4water3df))
type<-rep("root",times=nrow(roott1inoc4water3df))
roott1inoc4water3dffinal<-cbind(roott1inoc4water3df,water,inoculum,type)

#roott1inoc5water3
roott1inoc5water3df<-as.data.frame(roott1inoc5water3)
colnames(roott1inoc5water3df)[1] ="shannon"
water<-rep("water3",times=nrow(roott1inoc5water3df))
inoculum<-rep("inoc5",times=nrow(roott1inoc5water3df))
type<-rep("root",times=nrow(roott1inoc5water3df))
roott1inoc5water3dffinal<-cbind(roott1inoc5water3df,water,inoculum,type)

#roott1inoc6water3
roott1inoc6water3df<-as.data.frame(roott1inoc6water3)
colnames(roott1inoc6water3df)[1] ="shannon"
water<-rep("water3",times=nrow(roott1inoc6water3df))
inoculum<-rep("inoc6",times=nrow(roott1inoc6water3df))
type<-rep("root",times=nrow(roott1inoc6water3df))
roott1inoc6water3dffinal<-cbind(roott1inoc6water3df,water,inoculum,type)

#shoott1inoc1water1
shoott1inoc1water1df<-as.data.frame(shoott1inoc1water1)
colnames(shoott1inoc1water1df)[1] ="shannon"
water<-rep("water1",times=nrow(shoott1inoc1water1df))
inoculum<-rep("inoc1",times=nrow(shoott1inoc1water1df))
type<-rep("shoot",times=nrow(shoott1inoc1water1df))
shoott1inoc1water1dffinal<-cbind(shoott1inoc1water1df,water,inoculum,type)

#shoott1inoc4water1
shoott1inoc4water1df<-as.data.frame(shoott1inoc4water1)
colnames(shoott1inoc4water1df)[1] ="shannon"
water<-rep("water1",times=nrow(shoott1inoc4water1df))
inoculum<-rep("inoc4",times=nrow(shoott1inoc4water1df))
type<-rep("shoot",times=nrow(shoott1inoc4water1df))
shoott1inoc4water1dffinal<-cbind(shoott1inoc4water1df,water,inoculum,type)

#shoott1inoc5water1
shoott1inoc5water1df<-as.data.frame(shoott1inoc5water1)
colnames(shoott1inoc5water1df)[1] ="shannon"
water<-rep("water1",times=nrow(shoott1inoc5water1df))
inoculum<-rep("inoc5",times=nrow(shoott1inoc5water1df))
type<-rep("shoot",times=nrow(shoott1inoc5water1df))
shoott1inoc5water1dffinal<-cbind(shoott1inoc5water1df,water,inoculum,type)

#shoott1inoc6water1
shoott1inoc6water1df<-as.data.frame(shoott1inoc6water1)
colnames(shoott1inoc6water1df)[1] ="shannon"
water<-rep("water1",times=nrow(shoott1inoc6water1df))
inoculum<-rep("inoc6",times=nrow(shoott1inoc6water1df))
type<-rep("shoot",times=nrow(shoott1inoc6water1df))
shoott1inoc6water1dffinal<-cbind(shoott1inoc6water1df,water,inoculum,type)

#shoott1inoc1water2
shoott1inoc1water2df<-as.data.frame(shoott1inoc1water2)
colnames(shoott1inoc1water2df)[1] ="shannon"
water<-rep("water2",times=nrow(shoott1inoc1water2df))
inoculum<-rep("inoc1",times=nrow(shoott1inoc1water2df))
type<-rep("shoot",times=nrow(shoott1inoc1water2df))
shoott1inoc1water2dffinal<-cbind(shoott1inoc1water2df,water,inoculum,type)

#shoott1inoc4water2
shoott1inoc4water2df<-as.data.frame(shoott1inoc4water2)
colnames(shoott1inoc4water2df)[1] ="shannon"
water<-rep("water2",times=nrow(shoott1inoc4water2df))
inoculum<-rep("inoc4",times=nrow(shoott1inoc4water2df))
type<-rep("shoot",times=nrow(shoott1inoc4water2df))
shoott1inoc4water2dffinal<-cbind(shoott1inoc4water2df,water,inoculum,type)

#shoott1inoc5water2
shoott1inoc5water2df<-as.data.frame(shoott1inoc5water2)
colnames(shoott1inoc5water2df)[1] ="shannon"
water<-rep("water2",times=nrow(shoott1inoc5water2df))
inoculum<-rep("inoc5",times=nrow(shoott1inoc5water2df))
type<-rep("shoot",times=nrow(shoott1inoc5water2df))
shoott1inoc5water2dffinal<-cbind(shoott1inoc5water2df,water,inoculum,type)

#shoott1inoc6water2
shoott1inoc6water2df<-as.data.frame(shoott1inoc6water2)
colnames(shoott1inoc6water2df)[1] ="shannon"
water<-rep("water2",times=nrow(shoott1inoc6water2df))
inoculum<-rep("inoc6",times=nrow(shoott1inoc6water2df))
type<-rep("shoot",times=nrow(shoott1inoc6water2df))
shoott1inoc6water2dffinal<-cbind(shoott1inoc6water2df,water,inoculum,type)

#shoott1inoc1water3
shoott1inoc1water3df<-as.data.frame(shoott1inoc1water3)
colnames(shoott1inoc1water3df)[1] ="shannon"
water<-rep("water3",times=nrow(shoott1inoc1water3df))
inoculum<-rep("inoc1",times=nrow(shoott1inoc1water3df))
type<-rep("shoot",times=nrow(shoott1inoc1water3df))
shoott1inoc1water3dffinal<-cbind(shoott1inoc1water3df,water,inoculum,type)

#shoott1inoc4water3
shoott1inoc4water3df<-as.data.frame(shoott1inoc4water3)
colnames(shoott1inoc4water3df)[1] ="shannon"
water<-rep("water3",times=nrow(shoott1inoc4water3df))
inoculum<-rep("inoc4",times=nrow(shoott1inoc4water3df))
type<-rep("shoot",times=nrow(shoott1inoc4water3df))
shoott1inoc4water3dffinal<-cbind(shoott1inoc4water3df,water,inoculum,type)

#shoott1inoc5water3
shoott1inoc5water3df<-as.data.frame(shoott1inoc5water3)
colnames(shoott1inoc5water3df)[1] ="shannon"
water<-rep("water3",times=nrow(shoott1inoc5water3df))
inoculum<-rep("inoc5",times=nrow(shoott1inoc5water3df))
type<-rep("shoot",times=nrow(shoott1inoc5water3df))
shoott1inoc5water3dffinal<-cbind(shoott1inoc5water3df,water,inoculum,type)

#shoott1inoc6water3
shoott1inoc6water3df<-as.data.frame(shoott1inoc6water3)
colnames(shoott1inoc6water3df)[1] ="shannon"
water<-rep("water3",times=nrow(shoott1inoc6water3df))
inoculum<-rep("inoc6",times=nrow(shoott1inoc6water3df))
type<-rep("shoot",times=nrow(shoott1inoc6water3df))
shoott1inoc6water3dffinal<-cbind(shoott1inoc6water3df,water,inoculum,type)

#shoottinoc1
shoott0inoc1df<-as.data.frame(shoott0inoc1)
colnames(shoott0inoc1df)[1] ="shannon"
inoculum<-rep("inoc1",times=nrow(shoott0inoc1df))
shoott0inoc1dffinal<-cbind(shoott0inoc1df,inoculum)

#shoottinoc4
shoott0inoc4df<-as.data.frame(shoott0inoc4)
colnames(shoott0inoc4df)[1] ="shannon"
inoculum<-rep("inoc4",times=nrow(shoott0inoc4df))
shoott0inoc4dffinal<-cbind(shoott0inoc4df,inoculum)

#shoottinoc5
shoott0inoc5df<-as.data.frame(shoott0inoc5)
colnames(shoott0inoc5df)[1] ="shannon"
inoculum<-rep("inoc5",times=nrow(shoott0inoc5df))
shoott0inoc5dffinal<-cbind(shoott0inoc5df,inoculum)

#shoottinoc6
shoott0inoc6df<-as.data.frame(shoott0inoc6)
colnames(shoott0inoc6df)[1] ="shannon"
inoculum<-rep("inoc6",times=nrow(shoott0inoc6df))
shoott0inoc6dffinal<-cbind(shoott0inoc6df,inoculum)

#now rbind them together
shanfinalbindt0<-do.call("rbind",list(shoott0inoc1dffinal,shoott0inoc4dffinal,shoott0inoc5dffinal,shoott0inoc6dffinal))

shanfinalbindt1<-do.call("rbind",list(roott1inoc1water1dffinal,roott1inoc4water1dffinal,roott1inoc5water1dffinal,roott1inoc6water1dffinal,roott1inoc1water2dffinal,roott1inoc4water2dffinal,roott1inoc5water2dffinal,roott1inoc6water2dffinal,roott1inoc1water3dffinal,roott1inoc4water3dffinal,roott1inoc5water3dffinal,roott1inoc6water3dffinal,shoott1inoc1water1dffinal,shoott1inoc4water1dffinal,shoott1inoc5water1dffinal,shoott1inoc6water1dffinal,shoott1inoc1water2dffinal,shoott1inoc4water2dffinal,shoott1inoc5water2dffinal,shoott1inoc6water2dffinal,shoott1inoc1water3dffinal,shoott1inoc4water3dffinal,shoott1inoc5water3dffinal,shoott1inoc6water3dffinal))

# add in time
shanfinalbindt0$time.point<-"T0"
shanfinalbindt0$type<-"shoot"
shanfinalbindt0$water<-"T0"

# rearrange
col_order <- c("shannon", "water", "inoculum",
               "type", "time.point")
shanfinalbindt0 <- shanfinalbindt0[, col_order]

# add time for 
shanfinalbindt1$time.point<-"T1"

# full shannon df
full.shannon.df<-rbind(shanfinalbindt0,shanfinalbindt1)

# make factors
full.shannon.df$inoculum<-as.factor(full.shannon.df$inoculum)
levels(full.shannon.df$inoculum) <- list(
  Dry="inoc1", Moderate="inoc4", Wet="inoc5", Control="inoc6")

full.shannon.df$water<-as.factor(full.shannon.df$water)
levels(full.shannon.df$water) <- list(
  T0= "T0", Low="water1", Moderate="water2", High="water3")

```

#### Pooled Shannon's
```{r T0 and T1 all sample shannon}
# 'full.shannon.df' full df

# make interaction column for lattice
full.shannon.df$time.type<-interaction(full.shannon.df$time.point, full.shannon.df$type)

# set levels
full.shannon.df$time.type<-factor(full.shannon.df$time.type, 
                                  levels=c(
                                    "T0.shoot", 
                                    "T1.shoot", "T1.root"))

# make interaction column for colors
full.shannon.df$time.water<-interaction(full.shannon.df$water, full.shannon.df$time.point)

# set levels
full.shannon.df$time.water<-factor(full.shannon.df$time.water, 
                                  levels=c(
                                    "T0.T0", 
                                    "Low.T1", 
                                    "Moderate.T1", "High.T1"))


######### 3 panel
T0.shoot.shannon<-full.shannon.df[(full.shannon.df$time.type=="T0.shoot"),]
T1.shoot.shannon<-full.shannon.df[(full.shannon.df$time.type=="T1.shoot"),]
T1.root.shannon<-full.shannon.df[(full.shannon.df$time.type=="T1.root"),]

# T0 shoot
shannonviolin.T0.shoot<-ggplot(T0.shoot.shannon,
                               aes(x=water,y=shannon,fill=inoculum,color=inoculum)) + 
  geom_violin(alpha=0.2)+
  geom_point(position=position_dodge(0.9), alpha=0.7, size=0.3)+
  ylim(c(0,4.5))+
  scale_color_brewer(palette="Dark2")+
  scale_fill_brewer(palette="Dark2")+
  theme(
    axis.text = element_text(color='black',size=8),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank())

# T1 shoot
shannonviolin.T1.shoot<-ggplot(T1.shoot.shannon,
                               aes(x=water,y=shannon,fill=inoculum,color=inoculum)) + 
  geom_violin(position=position_dodge(0.7), alpha=0.2)+
  geom_point(position=position_dodge(0.7), alpha=0.7, size=0.3)+
  scale_color_brewer(palette="Dark2")+
  scale_fill_brewer(palette="Dark2")+
  ylim(c(0,4.5))+
  theme(
    axis.text = element_text(color='black',size=8),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank())

# T1 root
shannonviolin.T1.root<-ggplot(T1.root.shannon,
                               aes(x=water,y=shannon,fill=inoculum,color=inoculum)) + 
  geom_violin(position=position_dodge(0.7), alpha=0.2)+
  geom_point(position=position_dodge(0.7), alpha=0.7, size=0.3)+
  ylim(c(0,4.5))+
  scale_color_brewer(palette="Dark2")+
  scale_fill_brewer(palette="Dark2")+
  theme(
    axis.text = element_text(color='black',size=8),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank())


shan.legend <- get_legend(
  shannonviolin.T0.shoot + 
    theme(legend.box.margin = margin(0, 0, 0, 12)) +
    theme(legend.title=element_blank())+ # no need for a title above legend
    theme(legend.key.size = unit(0.5, "cm")))

shann_plots<-plot_grid(shannonviolin.T0.shoot + theme(legend.position = "none"),
                        shannonviolin.T1.shoot + theme(legend.position = "none"),
                        shannonviolin.T1.root + theme(legend.position = "none"),
                       shan.legend,
                       labels=c('A', 'B', 'C', ''), label_size=8, hjust=-1, vjust= 2, 
          rel_widths = c(4,8,8,2), ncol=4)

shann_plots
dev.copy(pdf, "figures/FigS5.shannon.pdf", width=10, height=4.5, encod="MacRoman")
dev.off()

```

#### Stats for Shannon
- this is table S5
```{r shannon stats}
Shannon.T0.shoot<-full.shannon.df[(full.shannon.df$time.type=="T0.shoot"),]
Shannon.T1.shoot<-full.shannon.df[(full.shannon.df$time.type=="T1.shoot"),]
Shannon.T1.root<-full.shannon.df[(full.shannon.df$time.type=="T1.root"),]

#### T0 shoots
T0.shoot.shan.mod<-lm(shannon~inoculum, Shannon.T0.shoot)
Anova(T0.shoot.shan.mod) # no effects

#### T1 shoot
T1.shoot.shan.mod<-lm(shannon~water*inoculum, Shannon.T1.shoot)
Anova(T1.shoot.shan.mod) # water and inoc effects

# interaction
posthoc.T1.shoot.a<-emmeans(T1.shoot.shan.mod,pairwise~inoculum|water)
multcomp::cld(posthoc.T1.shoot.a, Letters=letters)

#water
posthoc.T1.shoot.b<-emmeans(T1.shoot.shan.mod,pairwise~water)
multcomp::cld(posthoc.T1.shoot.b, Letters=letters)

#inoculum
posthoc.T1.shoot.c<-emmeans(T1.shoot.shan.mod,pairwise~inoculum)
multcomp::cld(posthoc.T1.shoot.c, Letters=letters)

##### T1 root
T1.root.shan.mod<-lm(shannon~water*inoculum, Shannon.T1.root)
Anova(T1.root.shan.mod) #  inoc effects

posthoc.T1.root<-emmeans(T1.root.shan.mod,pairwise~inoculum)
multcomp::cld(posthoc.T1.root, Letters=letters)

```


### Relative abundance  

```{r relative abundance in phyloseq, eval=FALSE}
# input.files, note that controls and slurry 2 and 3 would still be here
ASV.table<-read.delim("data/ASV_table_postfiltering.txt")
colnames(ASV.table)[colnames(ASV.table)=="X.OTU.ID"]<-"OTU.ID"

# adjust for rownames = OTU/ASVs
ASV.table <- ASV.table %>%
    tibble::column_to_rownames("OTU.ID") 

# remove taxonomy here for OTU table
ASV.table.edit<-within(ASV.table, rm(taxonomy))
OTU = otu_table(ASV.table.edit, taxa_are_rows = TRUE)

#### buildtaxonomy...
taxon<-as.data.frame(ASV.table$taxonomy)
rownames(taxon)<-rownames(ASV.table)
colnames(taxon)[colnames(taxon)=="ASV.table$taxonomy"]<-"taxonomy"

taxon.df<-tidyr::separate(taxon, taxonomy, into = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep = ";", fill = "right", extra = "drop")

# remove labels on names
taxon.df$Domain <- gsub('[k__]','',taxon.df$Domain)
taxon.df$Phylum <- gsub('[p__]','',taxon.df$Phylum)
taxon.df$Class <- gsub('[c__]','',taxon.df$Class)
taxon.df$Genus <- gsub('[g__]','',taxon.df$Genus)
taxon.df$Species <- gsub('[s__]','',taxon.df$Species)

taxon.mat<-as.matrix(taxon.df)
TAX<-tax_table(taxon.mat)


#### load in samples
samples_df<-read.delim("data/map_loaded_strawberries.txt")
colnames(samples_df)[colnames(samples_df)=="X.OTU.ID"]<-"OTU.ID"

samples_df$Water<- dplyr::recode_factor(samples_df$Water,
                "1" = "Low",
                "2" = "Moderate",
                "3" = "High",
                "NA"= "NA")

samples_df$Inoculum<- dplyr::recode_factor(samples_df$Inoculum,
                "1" = "Dry",
                "4" = "Moderate",
                "5" = "Wet",
                "6"= "Control")

samples_df$Trt<-interaction(samples_df$Water, samples_df$Inoculum)

# adjust for rownames = OTU/ASVs
samples_df <- samples_df %>%
    tibble::column_to_rownames("OTU.ID") 

# make phyloseq data
samples_data<-sample_data(samples_df)

### final objects
TAX
OTU
samples_data

# phloseq merged
phy.straw<-phyloseq(OTU, TAX,samples_data)

# rank by Genus
pseq_genus <- phyloseq::tax_glom(phy.straw, taxrank = "Family")

####### look at DE2seq
#install.packages("remotes")
#remotes::install_github("FelixErnst/mia")

################
# try DESeq2
#BiocManager::install("DESeq2")
library("DESeq2")

#drop control samples
phy.straw.nocontrol = subset_samples(phy.straw, TimePoint != "Control") #534 samples

#drop inoc 2 and 3
phy.straw.drop.inoc = subset_samples(phy.straw.nocontrol, 
                                     Inoculum != "2" & Inoculum != "3") # 346 samples

unique(tax_table(phy.straw.drop.inoc)
#prune 0 counts
phy.straw.drop.inoc<-prune_taxa(taxa_sums(phy.straw.drop.inoc)> 0, phy.straw.drop.inoc)

write_phyloseq(phy.straw.drop.inoc, "output/")

phy.straw.sh = subset_samples(phy.straw.drop.inoc, Type == "Shoot") #203 samples
phy.straw.rt = subset_samples(phy.straw.drop.inoc, Type == "Root") #143 samples

# roots 
phyDES.roots= phyloseq_to_deseq2(phy.straw.rt, ~Trt)
dds <- phyDES.roots[ rowSums(counts(phyDES.roots)) > 5, ]
counts(dds)
geoMeans <- apply(cts, 1, function(row) if (all(row == 0)) 0 else exp(sum(log(row[row != 0]))/length(row)))
dds <- estimateSizeFactors(dds, geoMeans=geoMeans)


table(colSums(counts(dds) == 0))
which.max(colSums(counts(dds) == 0))

diagdds = DESeq(dds2, test="Wald", fitType="parametric")


rs <- rowSums( counts(phyDES.roots) == 0 )
table(rs)

#1.Estimate Size Factors
counts<-counts(phyDES.roots)
geoMeans = apply(counts, 1, function(row) if (all(row == 0)) 0 else exp(mean(log(row[row != 0]))))
ddsLove<-estimateSizeFactors(phyDES.roots, geoMeans=geoMeans)
estimatesizefactors<-as.matrix(counts(ddsLove, normalized=TRUE)) 

#2. Estimate Dispersions
ddsLovedis<-estimateDispersions(ddsLove)
estimatedispersions<-as.data.frame(counts(ddsLovedis, normalized=TRUE))


install.packages("remotes")
remotes::install_github("FrederickHuangLin/ANCOMBC")
library(ANCOMBC)

out = ancombc(
  phyloseq = pseq_genus, 
  formula = "patient_status", 
  p_adj_method = "BH", 
  zero_cut = 0.90, # by default prevalence filter of 10% is applied
  lib_cut = 0, 
  group = "patient_status", 
  struc_zero = TRUE, 
  neg_lb = TRUE, 
  tol = 1e-5, 
  max_iter = 100, 
  conserve = TRUE, 
  alpha = 0.05, 
  global = TRUE
)
res <- out$res

```

Let's make some stacked bar plots too:
```{r relative abundance, results='hide'}
# let's refilter again to just root t1, shoot t1, and shoot t0
# Pair down the other data. We will have to hellinger transform this data too
# 'ASV.data' is the final df

## data subset already above
#ASV.T0.sh  # 62 samples, T0 shoots
#ASV.T1  # 284 samples, T1 all data
#ASV.T1.rt  # 143 samples, T1 root
#ASV.T1.sh # 203 samples, T1 shoots


#### Time 1
# all T1
inputstep5 = filter_data(ASV.data, 'TimePoint', keep_vals = 'T1')
# roots
roott1 = filter_data(inputstep5, 'Type', keep_vals = 'Root')
#shoots
shoott1 = filter_data(inputstep5, 'Type', keep_vals = 'Shoot')

#### Time 0
time0 = filter_data(ASV.data,'TimePoint',keep_vals = 'T0')
shoott0 = filter_data(time0, 'Type', keep_vals = 'Shoot')


#transformation
library(labdsv)
otuTAB<-shoott1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
shoott1$data_loaded<-t(hell)

otuTAB<-roott1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
roott1$data_loaded<-t(hell)

otuTAB<-shoott0$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
shoott0$data_loaded<-t(hell)
```

We will need to make a large dataframe here. View after melting to see
how many columns to make. This will be family level

```{r,results='hide'}
library(reshape2)
#level=5 is family. 
shoott1_tax_sum_family = summarize_taxonomy(shoott1, level = 5, report_higher_tax = FALSE)
shoott1_tax_sum_family$family <-row.names(shoott1_tax_sum_family)
shoott1_melt<- melt(shoott1_tax_sum_family, id.var="family")
nrow(shoott1_melt)
treatment<-rep("shoot_t1",times=nrow(shoott1_melt))
shoott1_melt<-cbind(shoott1_melt,treatment)

roott1_tax_sum_family = summarize_taxonomy(roott1, level = 5, report_higher_tax = FALSE)
roott1_tax_sum_family$family <-row.names(roott1_tax_sum_family)
roott1_melt<- melt(roott1_tax_sum_family, id.var="family")
nrow(roott1_melt)
treatment<-rep("root_t1",times=nrow(roott1_melt))
roott1_melt<-cbind(roott1_melt,treatment)

shoott0_tax_sum_family = summarize_taxonomy(shoott0, level = 5, report_higher_tax = FALSE)
shoott0_tax_sum_family$family <-row.names(shoott0_tax_sum_family)
shoott0_melt<- melt(shoott0_tax_sum_family, id.var="family")
nrow(shoott0_melt)
treatment<-rep("shoot_t0",times=nrow(shoott0_melt))
shoott0_melt<-cbind(shoott0_melt,treatment)

familyfinalbind<-do.call("rbind",list(shoott1_melt,roott1_melt,shoott0_melt))
```

family graph

```{r}
library(ggthemes)
library(ggplot2)
familygraph<-ggplot(familyfinalbind, aes(x = treatment, y = value, fill = family)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(92)) + 
  ggtitle("Family Graphs")+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=18),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key.size = unit(0.1, 'mm'),
    plot.title=element_text(size=20,face="bold",vjust=2))

familygraph
ggsave("figures/familygraph.pdf", height=8, width=17)

#ggsave(filename = "familygraph.pdf",
#       plot = familygraph,
#       bg = "transparent",
#       width = 17, height = 8, units = "in",
#       dpi = 600)
```


Let's try class
```{r,results='hide'}
library(reshape2)
#level=4 is class. 
shoott1_tax_sum_class = summarize_taxonomy(shoott1, level = 4, report_higher_tax = FALSE)
shoott1_tax_sum_class$class <-row.names(shoott1_tax_sum_class)
shoott1_melt<- melt(shoott1_tax_sum_class, id.var="class")
nrow(shoott1_melt)
treatment<-rep("shoot_t1",times=nrow(shoott1_melt))
shoott1_melt<-cbind(shoott1_melt,treatment)

roott1_tax_sum_class = summarize_taxonomy(roott1, level = 4, report_higher_tax = FALSE)
roott1_tax_sum_class$class <-row.names(roott1_tax_sum_class)
roott1_melt<- melt(roott1_tax_sum_class, id.var="class")
nrow(roott1_melt)
treatment<-rep("root_t1",times=nrow(roott1_melt))
roott1_melt<-cbind(roott1_melt,treatment)

shoott0_tax_sum_class = summarize_taxonomy(shoott0, level = 4, report_higher_tax = FALSE)
shoott0_tax_sum_class$class <-row.names(shoott0_tax_sum_class)
shoott0_melt<- melt(shoott0_tax_sum_class, id.var="class")
nrow(shoott0_melt)
treatment<-rep("shoot_t0",times=nrow(shoott0_melt))
shoott0_melt<-cbind(shoott0_melt,treatment)

classfinalbind<-do.call("rbind",list(shoott1_melt,roott1_melt,shoott0_melt))
```

class graph

```{r}
library(ggthemes)
library(ggplot2)
classgraph<-ggplot(classfinalbind, aes(x = treatment, y = value, fill = class)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(51)) + 
  ggtitle("class Graphs")+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=18),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key.size = unit(0.5, 'mm'),
    plot.title=element_text(size=20,face="bold",vjust=2))

classgraph
ggsave("figures/classgraph.pdf", height=6, width=10)

#ggsave(filename = "classgraph.pdf",
#       plot = classgraph,
#       bg = "transparent",
#      width = 10, height = 6, units = "in",
#       dpi = 600)
```

Now we can make t1 shoots by inoculation and water treatment
```{r,results='hide'}
#let's refilter again just in case but include the control too
#Pair down the other data. We will have to hellinger transform this data too
library(mctoolsr)
library(ggplot2)
library(labdsv)
inputstep5 = filter_data(input, 'TimePoint', keep_vals = 'T1')
shoots = filter_data(inputstep5, 'Type', keep_vals = 'Shoot')
water1 = filter_data(shoots, 'Water', keep_vals = '1')
water1inoc1 = filter_data(water1, 'Inoculum', keep_vals = '1')
water1inoc4 = filter_data(water1, 'Inoculum', keep_vals = '4')
water1inoc5 = filter_data(water1, 'Inoculum', keep_vals = '5')
water1inoc6 = filter_data(water1, 'Inoculum', keep_vals = '6')

water2 = filter_data(inputstep5, 'Water', keep_vals = '2')
water2inoc1 = filter_data(water2, 'Inoculum', keep_vals = '1')
water2inoc4 = filter_data(water2, 'Inoculum', keep_vals = '4')
water2inoc5 = filter_data(water2, 'Inoculum', keep_vals = '5')
water2inoc6 = filter_data(water2, 'Inoculum', keep_vals = '6')

water3 = filter_data(inputstep5, 'Water', keep_vals = '3')
water3inoc1 = filter_data(water3, 'Inoculum', keep_vals = '1')
water3inoc4 = filter_data(water3, 'Inoculum', keep_vals = '4')
water3inoc5 = filter_data(water3, 'Inoculum', keep_vals = '5')
water3inoc6 = filter_data(water3, 'Inoculum', keep_vals = '6')

#transformation
otuTAB<-water1inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)

water1inoc1$data_loaded<-t(hell)
otuTAB<-water1inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)

water1inoc4$data_loaded<-t(hell)
otuTAB<-water1inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc5$data_loaded<-t(hell)

otuTAB<-water1inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc6$data_loaded<-t(hell)

otuTAB<-water2inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc1$data_loaded<-t(hell)

otuTAB<-water2inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc4$data_loaded<-t(hell)

otuTAB<-water2inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc5$data_loaded<-t(hell)

otuTAB<-water2inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc6$data_loaded<-t(hell)

otuTAB<-water3inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc1$data_loaded<-t(hell)

otuTAB<-water3inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc4$data_loaded<-t(hell)

otuTAB<-water3inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc5$data_loaded<-t(hell)

otuTAB<-water3inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc6$data_loaded<-t(hell)

```

We will need to make a large dataframe here. View after melting to see how many columns to make. This will be family for shoots
```{r,results='hide'}
library(reshape2)
water1inoc1_tax_sum_family = summarize_taxonomy(water1inoc1, level = 5, report_higher_tax = FALSE)
water1inoc1_tax_sum_family$family <-row.names(water1inoc1_tax_sum_family)
water1inoc1_melt<- melt(water1inoc1_tax_sum_family, id.var="family")
nrow(water1inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water1inoc1_melt))
water<-rep("water1",times=nrow(water1inoc1_melt))
water1inoc1_melt<-cbind(water1inoc1_melt,inoculum,water)

water1inoc4_tax_sum_family = summarize_taxonomy(water1inoc4, level = 5, report_higher_tax = FALSE)
water1inoc4_tax_sum_family$family <-row.names(water1inoc4_tax_sum_family)
water1inoc4_melt<- melt(water1inoc4_tax_sum_family, id.var="family")
nrow(water1inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water1inoc4_melt))
water<-rep("water1",times=nrow(water1inoc4_melt))
water1inoc4_melt<-cbind(water1inoc4_melt,inoculum,water)

water1inoc5_tax_sum_family = summarize_taxonomy(water1inoc5, level = 5, report_higher_tax = FALSE)
water1inoc5_tax_sum_family$family <-row.names(water1inoc5_tax_sum_family)
water1inoc5_melt<- melt(water1inoc5_tax_sum_family, id.var="family")
nrow(water1inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water1inoc5_melt))
water<-rep("water1",times=nrow(water1inoc5_melt))
water1inoc5_melt<-cbind(water1inoc5_melt,inoculum,water)

water1inoc6_tax_sum_family = summarize_taxonomy(water1inoc6, level = 5, report_higher_tax = FALSE)
water1inoc6_tax_sum_family$family <-row.names(water1inoc6_tax_sum_family)
water1inoc6_melt<- melt(water1inoc6_tax_sum_family, id.var="family")
nrow(water1inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water1inoc6_melt))
water<-rep("water1",times=nrow(water1inoc6_melt))
water1inoc6_melt<-cbind(water1inoc6_melt,inoculum,water)

water2inoc1_tax_sum_family = summarize_taxonomy(water2inoc1, level = 5, report_higher_tax = FALSE)
water2inoc1_tax_sum_family$family <-row.names(water2inoc1_tax_sum_family)
water2inoc1_melt<- melt(water2inoc1_tax_sum_family, id.var="family")
nrow(water2inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water2inoc1_melt))
water<-rep("water2",times=nrow(water2inoc1_melt))
water2inoc1_melt<-cbind(water2inoc1_melt,inoculum,water)

water2inoc4_tax_sum_family = summarize_taxonomy(water2inoc4, level = 5, report_higher_tax = FALSE)
water2inoc4_tax_sum_family$family <-row.names(water2inoc4_tax_sum_family)
water2inoc4_melt<- melt(water2inoc4_tax_sum_family, id.var="family")
nrow(water2inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water2inoc4_melt))
water<-rep("water2",times=nrow(water2inoc4_melt))
water2inoc4_melt<-cbind(water2inoc4_melt,inoculum,water)

water2inoc5_tax_sum_family = summarize_taxonomy(water2inoc5, level = 5, report_higher_tax = FALSE)
water2inoc5_tax_sum_family$family <-row.names(water2inoc5_tax_sum_family)
water2inoc5_melt<- melt(water2inoc5_tax_sum_family, id.var="family")
nrow(water2inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water2inoc5_melt))
water<-rep("water2",times=nrow(water2inoc5_melt))
water2inoc5_melt<-cbind(water2inoc5_melt,inoculum,water)

water2inoc6_tax_sum_family = summarize_taxonomy(water2inoc6, level = 5, report_higher_tax = FALSE)
water2inoc6_tax_sum_family$family <-row.names(water2inoc6_tax_sum_family)
water2inoc6_melt<- melt(water2inoc6_tax_sum_family, id.var="family")
nrow(water2inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water2inoc6_melt))
water<-rep("water2",times=nrow(water2inoc6_melt))
water2inoc6_melt<-cbind(water2inoc6_melt,inoculum,water)

water3inoc1_tax_sum_family = summarize_taxonomy(water3inoc1, level = 5, report_higher_tax = FALSE)
water3inoc1_tax_sum_family$family <-row.names(water3inoc1_tax_sum_family)
water3inoc1_melt<- melt(water3inoc1_tax_sum_family, id.var="family")
nrow(water3inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water3inoc1_melt))
water<-rep("water3",times=nrow(water3inoc1_melt))
water3inoc1_melt<-cbind(water3inoc1_melt,inoculum,water)

water3inoc4_tax_sum_family = summarize_taxonomy(water3inoc4, level = 5, report_higher_tax = FALSE)
water3inoc4_tax_sum_family$family <-row.names(water3inoc4_tax_sum_family)
water3inoc4_melt<- melt(water3inoc4_tax_sum_family, id.var="family")
nrow(water3inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water3inoc4_melt))
water<-rep("water3",times=nrow(water3inoc4_melt))
water3inoc4_melt<-cbind(water3inoc4_melt,inoculum,water)

water3inoc5_tax_sum_family = summarize_taxonomy(water3inoc5, level = 5, report_higher_tax = FALSE)
water3inoc5_tax_sum_family$family <-row.names(water3inoc5_tax_sum_family)
water3inoc5_melt<- melt(water3inoc5_tax_sum_family, id.var="family")
nrow(water3inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water3inoc5_melt))
water<-rep("water3",times=nrow(water3inoc5_melt))
water3inoc5_melt<-cbind(water3inoc5_melt,inoculum,water)

water3inoc6_tax_sum_family = summarize_taxonomy(water3inoc6, level = 5, report_higher_tax = FALSE)
water3inoc6_tax_sum_family$family <-row.names(water3inoc6_tax_sum_family)
water3inoc6_melt<- melt(water3inoc6_tax_sum_family, id.var="family")
nrow(water3inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water3inoc6_melt))
water<-rep("water3",times=nrow(water3inoc6_melt))
water3inoc6_melt<-cbind(water3inoc6_melt,inoculum,water)

familyfinalbind<-do.call("rbind",list(water1inoc1_melt,water1inoc4_melt,water1inoc5_melt,water1inoc6_melt,water2inoc1_melt,water2inoc4_melt,water2inoc5_melt,water2inoc6_melt,water3inoc1_melt,water3inoc4_melt,water3inoc5_melt,water3inoc6_melt))
```


#### Shoots family graph
```{r}
library(ggthemes)
shootsfamily<-ggplot(familyfinalbind, aes(x = inoculum, y = value, fill = family)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(90)) + 
  ggtitle("Shoots Family Graphs")+
  facet_grid(.~water,scales ="free",space="free")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=18),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key.size = unit(1, 'mm'),
    legend.text = element_text(color='black',size=4),
    plot.title=element_text(size=20,face="bold",vjust=2))

shootsfamily
ggsave("figures/shootsfamily.pdf", height=8, width=17)

#ggsave(filename = "shootsfamily.pdf",
#       plot = shootsfamily,
#       bg = "transparent",
#       width = 17, height = 8, units = "in",
#       dpi = 600)
```

Now we can make t1 roots by inoculation and water treatment
```{r,results='hide'}
#let's refilter again just in case but include the control too
#Pair down the other data. We will have to hellinger transform this data too

inputstep5 = filter_data(ASV.data, 'TimePoint', keep_vals = 'T1')

roots = filter_data(inputstep5, 'Type', keep_vals = 'Root')
water1 = filter_data(roots, 'Water', keep_vals = '1')
water1inoc1 = filter_data(water1, 'Inoculum', keep_vals = '1')
water1inoc4 = filter_data(water1, 'Inoculum', keep_vals = '4')
water1inoc5 = filter_data(water1, 'Inoculum', keep_vals = '5')
water1inoc6 = filter_data(water1, 'Inoculum', keep_vals = '6')

water2 = filter_data(inputstep5, 'Water', keep_vals = '2')
water2inoc1 = filter_data(water2, 'Inoculum', keep_vals = '1')
water2inoc4 = filter_data(water2, 'Inoculum', keep_vals = '4')
water2inoc5 = filter_data(water2, 'Inoculum', keep_vals = '5')
water2inoc6 = filter_data(water2, 'Inoculum', keep_vals = '6')

water3 = filter_data(inputstep5, 'Water', keep_vals = '3')
water3inoc1 = filter_data(water3, 'Inoculum', keep_vals = '1')
water3inoc4 = filter_data(water3, 'Inoculum', keep_vals = '4')
water3inoc5 = filter_data(water3, 'Inoculum', keep_vals = '5')
water3inoc6 = filter_data(water3, 'Inoculum', keep_vals = '6')

#transformation
otuTAB<-water1inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc1$data_loaded<-t(hell)

otuTAB<-water1inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc4$data_loaded<-t(hell)

otuTAB<-water1inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc5$data_loaded<-t(hell)

otuTAB<-water1inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc6$data_loaded<-t(hell)

otuTAB<-water2inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc1$data_loaded<-t(hell)

otuTAB<-water2inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc4$data_loaded<-t(hell)

otuTAB<-water2inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc5$data_loaded<-t(hell)

otuTAB<-water2inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water2inoc6$data_loaded<-t(hell)

otuTAB<-water3inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc1$data_loaded<-t(hell)

otuTAB<-water3inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc4$data_loaded<-t(hell)

otuTAB<-water3inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc5$data_loaded<-t(hell)

otuTAB<-water3inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water3inoc6$data_loaded<-t(hell)

```

We will need to make a large dataframe here. View after melting to see how many columns to make. This will be family for shoots
```{r,results='hide'}
library(reshape2)
water1inoc1_tax_sum_family = summarize_taxonomy(water1inoc1, level = 5, report_higher_tax = FALSE)
water1inoc1_tax_sum_family$family <-row.names(water1inoc1_tax_sum_family)
water1inoc1_melt<- melt(water1inoc1_tax_sum_family, id.var="family")
nrow(water1inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water1inoc1_melt))
water<-rep("water1",times=nrow(water1inoc1_melt))
water1inoc1_melt<-cbind(water1inoc1_melt,inoculum,water)

water1inoc4_tax_sum_family = summarize_taxonomy(water1inoc4, level = 5, report_higher_tax = FALSE)
water1inoc4_tax_sum_family$family <-row.names(water1inoc4_tax_sum_family)
water1inoc4_melt<- melt(water1inoc4_tax_sum_family, id.var="family")
nrow(water1inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water1inoc4_melt))
water<-rep("water1",times=nrow(water1inoc4_melt))
water1inoc4_melt<-cbind(water1inoc4_melt,inoculum,water)

water1inoc5_tax_sum_family = summarize_taxonomy(water1inoc5, level = 5, report_higher_tax = FALSE)
water1inoc5_tax_sum_family$family <-row.names(water1inoc5_tax_sum_family)
water1inoc5_melt<- melt(water1inoc5_tax_sum_family, id.var="family")
nrow(water1inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water1inoc5_melt))
water<-rep("water1",times=nrow(water1inoc5_melt))
water1inoc5_melt<-cbind(water1inoc5_melt,inoculum,water)

water1inoc6_tax_sum_family = summarize_taxonomy(water1inoc6, level = 5, report_higher_tax = FALSE)
water1inoc6_tax_sum_family$family <-row.names(water1inoc6_tax_sum_family)
water1inoc6_melt<- melt(water1inoc6_tax_sum_family, id.var="family")
nrow(water1inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water1inoc6_melt))
water<-rep("water1",times=nrow(water1inoc6_melt))
water1inoc6_melt<-cbind(water1inoc6_melt,inoculum,water)

water2inoc1_tax_sum_family = summarize_taxonomy(water2inoc1, level = 5, report_higher_tax = FALSE)
water2inoc1_tax_sum_family$family <-row.names(water2inoc1_tax_sum_family)
water2inoc1_melt<- melt(water2inoc1_tax_sum_family, id.var="family")
nrow(water2inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water2inoc1_melt))
water<-rep("water2",times=nrow(water2inoc1_melt))
water2inoc1_melt<-cbind(water2inoc1_melt,inoculum,water)

water2inoc4_tax_sum_family = summarize_taxonomy(water2inoc4, level = 5, report_higher_tax = FALSE)
water2inoc4_tax_sum_family$family <-row.names(water2inoc4_tax_sum_family)
water2inoc4_melt<- melt(water2inoc4_tax_sum_family, id.var="family")
nrow(water2inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water2inoc4_melt))
water<-rep("water2",times=nrow(water2inoc4_melt))
water2inoc4_melt<-cbind(water2inoc4_melt,inoculum,water)

water2inoc5_tax_sum_family = summarize_taxonomy(water2inoc5, level = 5, report_higher_tax = FALSE)
water2inoc5_tax_sum_family$family <-row.names(water2inoc5_tax_sum_family)
water2inoc5_melt<- melt(water2inoc5_tax_sum_family, id.var="family")
nrow(water2inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water2inoc5_melt))
water<-rep("water2",times=nrow(water2inoc5_melt))
water2inoc5_melt<-cbind(water2inoc5_melt,inoculum,water)

water2inoc6_tax_sum_family = summarize_taxonomy(water2inoc6, level = 5, report_higher_tax = FALSE)
water2inoc6_tax_sum_family$family <-row.names(water2inoc6_tax_sum_family)
water2inoc6_melt<- melt(water2inoc6_tax_sum_family, id.var="family")
nrow(water2inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water2inoc6_melt))
water<-rep("water2",times=nrow(water2inoc6_melt))
water2inoc6_melt<-cbind(water2inoc6_melt,inoculum,water)

water3inoc1_tax_sum_family = summarize_taxonomy(water3inoc1, level = 5, report_higher_tax = FALSE)
water3inoc1_tax_sum_family$family <-row.names(water3inoc1_tax_sum_family)
water3inoc1_melt<- melt(water3inoc1_tax_sum_family, id.var="family")
nrow(water3inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water3inoc1_melt))
water<-rep("water3",times=nrow(water3inoc1_melt))
water3inoc1_melt<-cbind(water3inoc1_melt,inoculum,water)

water3inoc4_tax_sum_family = summarize_taxonomy(water3inoc4, level = 5, report_higher_tax = FALSE)
water3inoc4_tax_sum_family$family <-row.names(water3inoc4_tax_sum_family)
water3inoc4_melt<- melt(water3inoc4_tax_sum_family, id.var="family")
nrow(water3inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water3inoc4_melt))
water<-rep("water3",times=nrow(water3inoc4_melt))
water3inoc4_melt<-cbind(water3inoc4_melt,inoculum,water)

water3inoc5_tax_sum_family = summarize_taxonomy(water3inoc5, level = 5, report_higher_tax = FALSE)
water3inoc5_tax_sum_family$family <-row.names(water3inoc5_tax_sum_family)
water3inoc5_melt<- melt(water3inoc5_tax_sum_family, id.var="family")
nrow(water3inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water3inoc5_melt))
water<-rep("water3",times=nrow(water3inoc5_melt))
water3inoc5_melt<-cbind(water3inoc5_melt,inoculum,water)

water3inoc6_tax_sum_family = summarize_taxonomy(water3inoc6, level = 5, report_higher_tax = FALSE)
water3inoc6_tax_sum_family$family <-row.names(water3inoc6_tax_sum_family)
water3inoc6_melt<- melt(water3inoc6_tax_sum_family, id.var="family")
nrow(water3inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water3inoc6_melt))
water<-rep("water3",times=nrow(water3inoc6_melt))
water3inoc6_melt<-cbind(water3inoc6_melt,inoculum,water)

familyrootfinalbind<-do.call("rbind",list(water1inoc1_melt,water1inoc4_melt,water1inoc5_melt,water1inoc6_melt,water2inoc1_melt,water2inoc4_melt,water2inoc5_melt,water2inoc6_melt,water3inoc1_melt,water3inoc4_melt,water3inoc5_melt,water3inoc6_melt))
```


#### Roots family graph
```{r dataframes}
library(ggthemes)
Rootsfamily<-ggplot(familyrootfinalbind, aes(x = inoculum, y = value, fill = family)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(90)) + 
  ggtitle("Roots Family Graphs")+
  facet_grid(.~water,scales ="free",space="free")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=18),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key.size = unit(1, 'mm'),
    legend.text = element_text(color='black',size=4),
    plot.title=element_text(size=20,face="bold",vjust=2))

Rootsfamily
ggsave("figures/Rootsfamily.pdf", height=8, width=17)
```

Now we can make t0 shoots by inoculation and water treatment
```{r,results='hide'}
#let's refilter again just in case but include the control too
#Pair down the other data. We will have to hellinger transform this data too

inputstep5 = filter_data(input, 'TimePoint', keep_vals = 'T0')
shoots = filter_data(inputstep5, 'Type', keep_vals = 'Shoot')
water1inoc1 = filter_data(shoots, 'Inoculum', keep_vals = '1')
water1inoc4 = filter_data(shoots, 'Inoculum', keep_vals = '4')
water1inoc5 = filter_data(shoots, 'Inoculum', keep_vals = '5')
water1inoc6 = filter_data(shoots, 'Inoculum', keep_vals = '6')


#transformation
otuTAB<-water1inoc1$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc1$data_loaded<-t(hell)

otuTAB<-water1inoc4$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc4$data_loaded<-t(hell)

otuTAB<-water1inoc5$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc5$data_loaded<-t(hell)

otuTAB<-water1inoc6$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
water1inoc6$data_loaded<-t(hell)
```

We will need to make a large dataframe here. View after melting to see how many columns to make. This will be family for shoots t0
```{r,results='hide'}
library(reshape2)
water1inoc1_tax_sum_family = summarize_taxonomy(water1inoc1, level = 5, report_higher_tax = FALSE)
water1inoc1_tax_sum_family$family <-row.names(water1inoc1_tax_sum_family)
water1inoc1_melt<- melt(water1inoc1_tax_sum_family, id.var="family")
nrow(water1inoc1_melt)
inoculum<-rep("inoc1",times=nrow(water1inoc1_melt))
water<-rep("water1",times=nrow(water1inoc1_melt))
water1inoc1_melt<-cbind(water1inoc1_melt,inoculum,water)

water1inoc4_tax_sum_family = summarize_taxonomy(water1inoc4, level = 5, report_higher_tax = FALSE)
water1inoc4_tax_sum_family$family <-row.names(water1inoc4_tax_sum_family)
water1inoc4_melt<- melt(water1inoc4_tax_sum_family, id.var="family")
nrow(water1inoc4_melt)
inoculum<-rep("inoc4",times=nrow(water1inoc4_melt))
water<-rep("water1",times=nrow(water1inoc4_melt))
water1inoc4_melt<-cbind(water1inoc4_melt,inoculum,water)

water1inoc5_tax_sum_family = summarize_taxonomy(water1inoc5, level = 5, report_higher_tax = FALSE)
water1inoc5_tax_sum_family$family <-row.names(water1inoc5_tax_sum_family)
water1inoc5_melt<- melt(water1inoc5_tax_sum_family, id.var="family")
nrow(water1inoc5_melt)
inoculum<-rep("inoc5",times=nrow(water1inoc5_melt))
water<-rep("water1",times=nrow(water1inoc5_melt))
water1inoc5_melt<-cbind(water1inoc5_melt,inoculum,water)

water1inoc6_tax_sum_family = summarize_taxonomy(water1inoc6, level = 5, report_higher_tax = FALSE)
water1inoc6_tax_sum_family$family <-row.names(water1inoc6_tax_sum_family)
water1inoc6_melt<- melt(water1inoc6_tax_sum_family, id.var="family")
nrow(water1inoc6_melt)
inoculum<-rep("inoc6",times=nrow(water1inoc6_melt))
water<-rep("water1",times=nrow(water1inoc6_melt))
water1inoc6_melt<-cbind(water1inoc6_melt,inoculum,water)

familyshoott0finalbind<-do.call("rbind",list(water1inoc1_melt,water1inoc4_melt,water1inoc5_melt,water1inoc6_melt))
```


#### Shoots T0 family graph
```{r}
library(ggthemes)

Shootst0family<-ggplot(familyshoott0finalbind, aes(x = inoculum, y = value, fill = family)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(90)) + 
  ggtitle("Shoots T0 Family Graphs")+
  facet_grid(.~water,scales ="free",space="free")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=18),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.key.size = unit(1, 'mm'),
    legend.text = element_text(color='black',size=4),
    plot.title=element_text(size=20,face="bold",vjust=2))

Shootst0family
ggsave("figures/Shootst0family.pdf", height=8, width=10)
```

#### Venn Diagram
Let's start with the raw data. The venn diagram we want is split into
the three water groups. Then we need original inoculum, leaves post-inoc
post stress, leaves post-inoc pre stress, roots post inoc post stress

Now we need to pair down the data. Then do nrows. Then merge by the rows
and then do nrows. Then we can make the euler diagrams.

##### T1 Shoots Venn
```{r Venn shoots T1}

##########
##########
### all data
ASV.otu.tab<-as.data.frame(t(ASV.data$data_loaded))
ASV.data.all<-as.data.frame(cbind(ASV.data$map_loaded,ASV.otu.tab))

# shoots
shoot.T1.otu<-as.data.frame(t(ASV.T1.sh$data_loaded))
shoot.T1.otu.dat<-as.data.frame(cbind(ASV.T1.sh$map_loaded,shoot.T1.otu))

# roots
root.T1.otu<-as.data.frame(t(ASV.T1.rt$data_loaded))
root.T1.otu.dat<-as.data.frame(cbind(ASV.T1.rt$map_loaded,root.T1.otu))


### shoot.T1.otu.dat

## #shoots by water
shoot.T1.W1<-shoot.T1.otu.dat[(shoot.T1.otu.dat$Water=="1"),]
shoot.T1.W2<-shoot.T1.otu.dat[(shoot.T1.otu.dat$Water=="2"),]
shoot.T1.W3<-shoot.T1.otu.dat[(shoot.T1.otu.dat$Water=="3"),]


# subset to get inoculum 
# WATER 1
# Inoc 1
OTU.W1.inoc1.sh = colnames(shoot.T1.W1[shoot.T1.W1$Inoculum == "1",
              apply(shoot.T1.W1[shoot.T1.W1$Inoculum == "1",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc1.sh.ASV<- OTU.W1.inoc1.sh[-1:-10] # remove factors info

# Inoc 4
OTU.W1.inoc4.sh = colnames(shoot.T1.W1[shoot.T1.W1$Inoculum == "4",
              apply(shoot.T1.W1[shoot.T1.W1$Inoculum == "4",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc4.sh.ASV<- OTU.W1.inoc4.sh[-1:-10] # remove factors info

# Inoc 5
OTU.W1.inoc5.sh = colnames(shoot.T1.W1[shoot.T1.W1$Inoculum == "5",
              apply(shoot.T1.W1[shoot.T1.W1$Inoculum == "5",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc5.sh.ASV<- OTU.W1.inoc5.sh[-1:-10] # remove factors info

# Inoc 6
OTU.W1.inoc6.sh = colnames(shoot.T1.W1[shoot.T1.W1$Inoculum == "6",
              apply(shoot.T1.W1[shoot.T1.W1$Inoculum == "6",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc6.sh.ASV<- OTU.W1.inoc6.sh[-1:-10] # remove factors info

ASV.list.W1<-list("inoc1"=OTU.W1.inoc1.sh.ASV, "inoc4"=OTU.W1.inoc4.sh.ASV, 
          "inoc5"=OTU.W1.inoc5.sh.ASV, "inoc6"=OTU.W1.inoc6.sh.ASV)

venn(list("inoc1"=OTU.W1.inoc1.sh.ASV, "inoc4"=OTU.W1.inoc4.sh.ASV, 
          "inoc5"=OTU.W1.inoc5.sh.ASV, "inoc6"=OTU.W1.inoc6.sh.ASV))

######### venn diagram for Water 1
Venn.sh.T1.W1<-ggVennDiagram(ASV.list.W1) + 
  scale_fill_distiller(palette = "YlOrRd", direction = 1)+
  ggtitle("Shoots: T1, Water1")


######### 

# WATER 2
# Inoc 1
OTU.W2.inoc1.sh = colnames(shoot.T1.W2[shoot.T1.W2$Inoculum == "1",
              apply(shoot.T1.W2[shoot.T1.W2$Inoculum == "1",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc1.sh.ASV<- OTU.W2.inoc1.sh[-1:-10] # remove factors info

# Inoc 4
OTU.W2.inoc4.sh = colnames(shoot.T1.W2[shoot.T1.W2$Inoculum == "4",
              apply(shoot.T1.W2[shoot.T1.W2$Inoculum == "4",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc4.sh.ASV<- OTU.W2.inoc4.sh[-1:-10] # remove factors info

# Inoc 5
OTU.W2.inoc5.sh = colnames(shoot.T1.W2[shoot.T1.W2$Inoculum == "5",
              apply(shoot.T1.W2[shoot.T1.W2$Inoculum == "5",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc5.sh.ASV<- OTU.W2.inoc5.sh[-1:-10] # remove factors info

# Inoc 6
OTU.W2.inoc6.sh = colnames(shoot.T1.W2[shoot.T1.W2$Inoculum == "6",
              apply(shoot.T1.W2[shoot.T1.W2$Inoculum == "6",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc6.sh.ASV<- OTU.W2.inoc6.sh[-1:-10] # remove factors info

ASV.list.W2<-list("inoc1"=OTU.W2.inoc1.sh.ASV, "inoc4"=OTU.W2.inoc4.sh.ASV, 
          "inoc5"=OTU.W2.inoc5.sh.ASV, "inoc6"=OTU.W2.inoc6.sh.ASV)

venn(list("inoc1"=OTU.W2.inoc1.sh.ASV, "inoc4"=OTU.W2.inoc4.sh.ASV, 
          "inoc5"=OTU.W2.inoc5.sh.ASV, "inoc6"=OTU.W2.inoc6.sh.ASV))

######### venn diagram for Water 2
Venn.sh.T1.W2<-ggVennDiagram(ASV.list.W2) + 
  scale_fill_distiller(palette = "YlOrRd", direction = 1)+
  ggtitle("Shoots: T1, Water2")
######### 


# WATER 3
# Inoc 1
OTU.W3.inoc1.sh = colnames(shoot.T1.W3[shoot.T1.W3$Inoculum == "1",
              apply(shoot.T1.W3[shoot.T1.W3$Inoculum == "1",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc1.sh.ASV<- OTU.W3.inoc1.sh[-1:-10] # remove factors info

# Inoc 4
OTU.W3.inoc4.sh = colnames(shoot.T1.W3[shoot.T1.W3$Inoculum == "4",
              apply(shoot.T1.W3[shoot.T1.W3$Inoculum == "4",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc4.sh.ASV<- OTU.W3.inoc4.sh[-1:-10] # remove factors info

# Inoc 5
OTU.W3.inoc5.sh = colnames(shoot.T1.W3[shoot.T1.W3$Inoculum == "5",
              apply(shoot.T1.W3[shoot.T1.W3$Inoculum == "5",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc5.sh.ASV<- OTU.W3.inoc5.sh[-1:-10] # remove factors info

# Inoc 6
OTU.W3.inoc6.sh = colnames(shoot.T1.W3[shoot.T1.W3$Inoculum == "6",
              apply(shoot.T1.W3[shoot.T1.W3$Inoculum == "6",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc6.sh.ASV<- OTU.W3.inoc6.sh[-1:-10] # remove factors info

ASV.list.W3<-list("inoc1"=OTU.W3.inoc1.sh.ASV, "inoc4"=OTU.W3.inoc4.sh.ASV, 
          "inoc5"=OTU.W3.inoc5.sh.ASV, "inoc6"=OTU.W3.inoc6.sh.ASV)

venn(list("inoc1"=OTU.W3.inoc1.sh.ASV, "inoc4"=OTU.W3.inoc4.sh.ASV, 
          "inoc5"=OTU.W3.inoc5.sh.ASV, "inoc6"=OTU.W3.inoc6.sh.ASV))


######### venn diagram for Water 3
Venn.sh.T1.W3<-ggVennDiagram(ASV.list.W3) + 
  scale_fill_distiller(palette = "YlOrRd", direction = 1)+
  ggtitle("Shoots: T1, Water3")


Venn.sh.T1<-plot_grid(Venn.sh.T1.W1, Venn.sh.T1.W2, Venn.sh.T1.W3, ncol=3)
Venn.sh.T1
dev.copy(pdf, "figures/Venn.sh.T1.pdf", width=20, height=8, encod="MacRoman")
dev.off()


```

##### T1 Roots Venn
```{r Venn roots T1}

##########
##########

# roots
root.T1.otu<-as.data.frame(t(ASV.T1.rt$data_loaded))
root.T1.otu.dat<-as.data.frame(cbind(ASV.T1.rt$map_loaded,root.T1.otu))

### root.T1.otu.dat

## #roots by water
root.T1.W1<-root.T1.otu.dat[(root.T1.otu.dat$Water=="1"),]
root.T1.W2<-root.T1.otu.dat[(root.T1.otu.dat$Water=="2"),]
root.T1.W3<-root.T1.otu.dat[(root.T1.otu.dat$Water=="3"),]

#### ROOOTS!!
# subset to get inoculum 
# WATER 1
# Inoc 1
OTU.W1.inoc1.rt = colnames(root.T1.W1[root.T1.W1$Inoculum == "1",
              apply(root.T1.W1[root.T1.W1$Inoculum == "1",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc1.rt.ASV<- OTU.W1.inoc1.rt[-1:-10] # remove factors info

# Inoc 4
OTU.W1.inoc4.rt = colnames(root.T1.W1[root.T1.W1$Inoculum == "4",
              apply(root.T1.W1[root.T1.W1$Inoculum == "4",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc4.rt.ASV<- OTU.W1.inoc4.rt[-1:-10] # remove factors info

# Inoc 5
OTU.W1.inoc5.rt = colnames(root.T1.W1[root.T1.W1$Inoculum == "5",
              apply(root.T1.W1[root.T1.W1$Inoculum == "5",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc5.rt.ASV<- OTU.W1.inoc5.rt[-1:-10] # remove factors info

# Inoc 6
OTU.W1.inoc6.rt = colnames(root.T1.W1[root.T1.W1$Inoculum == "6",
              apply(root.T1.W1[root.T1.W1$Inoculum == "6",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W1.inoc6.rt.ASV<- OTU.W1.inoc6.rt[-1:-10] # remove factors info

ASV.list.W1<-list("inoc1"=OTU.W1.inoc1.rt.ASV, "inoc4"=OTU.W1.inoc4.rt.ASV, 
          "inoc5"=OTU.W1.inoc5.rt.ASV, "inoc6"=OTU.W1.inoc6.rt.ASV)

venn(list("inoc1"=OTU.W1.inoc1.rt.ASV, "inoc4"=OTU.W1.inoc4.rt.ASV, 
          "inoc5"=OTU.W1.inoc5.rt.ASV, "inoc6"=OTU.W1.inoc6.rt.ASV))

######### venn diagram for Water 1
Venn.rt.T1.W1<-ggVennDiagram(ASV.list.W1) + 
  scale_fill_distiller(palette = "YlGnBu", direction = 1)+
  ggtitle("Roots: T1, Water1")


######### 

# WATER 2
# Inoc 1
OTU.W2.inoc1.rt = colnames(root.T1.W2[root.T1.W2$Inoculum == "1",
              apply(root.T1.W2[root.T1.W2$Inoculum == "1",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc1.rt.ASV<- OTU.W2.inoc1.rt[-1:-10] # remove factors info

# Inoc 4
OTU.W2.inoc4.rt = colnames(root.T1.W2[root.T1.W2$Inoculum == "4",
              apply(root.T1.W2[root.T1.W2$Inoculum == "4",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc4.rt.ASV<- OTU.W2.inoc4.rt[-1:-10] # remove factors info

# Inoc 5
OTU.W2.inoc5.rt = colnames(root.T1.W2[root.T1.W2$Inoculum == "5",
              apply(root.T1.W2[root.T1.W2$Inoculum == "5",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc5.rt.ASV<- OTU.W2.inoc5.rt[-1:-10] # remove factors info

# Inoc 6
OTU.W2.inoc6.rt = colnames(root.T1.W2[root.T1.W2$Inoculum == "6",
              apply(root.T1.W2[root.T1.W2$Inoculum == "6",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W2.inoc6.rt.ASV<- OTU.W2.inoc6.rt[-1:-10] # remove factors info

ASV.list.W2<-list("inoc1"=OTU.W2.inoc1.rt.ASV, "inoc4"=OTU.W2.inoc4.rt.ASV, 
          "inoc5"=OTU.W2.inoc5.rt.ASV, "inoc6"=OTU.W2.inoc6.rt.ASV)

venn(list("inoc1"=OTU.W2.inoc1.rt.ASV, "inoc4"=OTU.W2.inoc4.rt.ASV, 
          "inoc5"=OTU.W2.inoc5.rt.ASV, "inoc6"=OTU.W2.inoc6.rt.ASV))

######### venn diagram for Water 2
Venn.rt.T1.W2<-ggVennDiagram(ASV.list.W2) + 
  scale_fill_distiller(palette = "YlGnBu", direction = 1)+
  ggtitle("Roots: T1, Water2")
######### 


# WATER 3
# Inoc 1
OTU.W3.inoc1.rt = colnames(root.T1.W3[root.T1.W3$Inoculum == "1",
              apply(root.T1.W3[root.T1.W3$Inoculum == "1",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc1.rt.ASV<- OTU.W3.inoc1.rt[-1:-10] # remove factors info

# Inoc 4
OTU.W3.inoc4.rt = colnames(root.T1.W3[root.T1.W3$Inoculum == "4",
              apply(root.T1.W3[root.T1.W3$Inoculum == "4",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc4.rt.ASV<- OTU.W3.inoc4.rt[-1:-10] # remove factors info

# Inoc 5
OTU.W3.inoc5.rt = colnames(root.T1.W3[root.T1.W3$Inoculum == "5",
              apply(root.T1.W3[root.T1.W3$Inoculum == "5",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc5.rt.ASV<- OTU.W3.inoc5.rt[-1:-10] # remove factors info

# Inoc 6
OTU.W3.inoc6.rt = colnames(root.T1.W3[root.T1.W3$Inoculum == "6",
              apply(root.T1.W3[root.T1.W3$Inoculum == "6",], 
              MARGIN=2, function(x) any(x >0))])
OTU.W3.inoc6.rt.ASV<- OTU.W3.inoc6.rt[-1:-10] # remove factors info

ASV.list.W3<-list("inoc1"=OTU.W3.inoc1.rt.ASV, "inoc4"=OTU.W3.inoc4.rt.ASV, 
          "inoc5"=OTU.W3.inoc5.rt.ASV, "inoc6"=OTU.W3.inoc6.rt.ASV)

venn(list("inoc1"=OTU.W3.inoc1.rt.ASV, "inoc4"=OTU.W3.inoc4.rt.ASV, 
          "inoc5"=OTU.W3.inoc5.rt.ASV, "inoc6"=OTU.W3.inoc6.rt.ASV))

######### venn diagram for Water 3
Venn.rt.T1.W3<-ggVennDiagram(ASV.list.W3) + 
  scale_fill_distiller(palette = "YlGnBu", direction = 1)+
  ggtitle("Roots: T1, Water3")


Venn.rt.T1<-plot_grid(Venn.rt.T1.W1, Venn.rt.T1.W2, Venn.rt.T1.W3, ncol=3)
Venn.rt.T1
dev.copy(pdf, "figures/Venn.rt.T1.pdf", width=20, height=8, encod="MacRoman")
dev.off()


```



First let's split into inoculation 1, 4 and 5

##### Inoculum 1
```{r inoculum 1 loaded,results='hide'}
input <- load_taxa_table("data/ASV_table_postfiltering.txt", "data/map_loaded_strawberries.txt")

#inoc1
inputinoc1only = filter_data(input, 'Inoculum', keep_vals = '1')
#leaves post-inoc post stress
inputinoc1onlyshoot = filter_data(inputinoc1only, 'Type', keep_vals = 'Shoot')
inoc1shootpostinocpoststress = filter_data(inputinoc1onlyshoot, 'SampleType', keep_vals = 'post_inoc_post_stress')
i1spips<-as.data.frame(inoc1shootpostinocpoststress$data_loaded)
i1spips$ASV <-row.names(i1spips)

#roots post-inoc post-stress
inputinoc1onlyroot = filter_data(inputinoc1only, 'Type', keep_vals = 'Root')
inoc1rootpostinocpoststress = filter_data(inputinoc1onlyroot, 'SampleType', keep_vals = 'post_inoc_post_stress')
i1rpips<-as.data.frame(inoc1rootpostinocpoststress$data_loaded)
i1rpips$ASV <-row.names(i1rpips)

#leaves post-inoc pre-stress
inoc1shootpostinocprestress = filter_data(inputinoc1onlyshoot, 'SampleType', keep_vals = 'post_inoc_pre_stress')
i1spiprs<-as.data.frame(inoc1shootpostinocprestress$data_loaded)
i1spiprs$ASV <-row.names(i1spiprs)

#Then let's grab original inoculum. We will need to grab the original data frame again.
inputraw <- load_taxa_table("data/ASV_table_raw.txt", "data/strawberries_raw_full_data.txt")

slurryonly = filter_data(inputraw, 'SampleType', keep_vals = 'slurry')
slurryonlyinoc1 = filter_data(slurryonly, 'Type', keep_vals = 'Inoculum 1')
slurryinoc1<-as.data.frame(slurryonlyinoc1$data_loaded)
slurryinoc1$ASV <-row.names(slurryinoc1)

```

Let's count the number of rows and assess overlap:
```{r}
nrow(i1spips)
nrow(i1rpips)
nrow(i1spiprs)
nrow(slurryinoc1)

##### Let's see how these all overlap:

i1spips_i1rpips <- inner_join(i1spips, i1rpips, 
                       c("ASV"))
nrow(i1spips_i1rpips)
i1spips_i1spiprs <- inner_join(i1spips, i1spiprs, 
                       c("ASV"))
nrow(i1spips_i1spiprs)
i1spips_slurry <- inner_join(i1spips, slurryinoc1, 
                       c("ASV"))
nrow(i1spips_slurry)
i1rpips_i1spiprs <- inner_join(i1spiprs, i1rpips, 
                       c("ASV"))
nrow(i1rpips_i1spiprs)
i1rpips_slurry <- inner_join(slurryinoc1, i1rpips, 
                       c("ASV"))
nrow(i1rpips_slurry)
i1spiprs_slurry <- inner_join(slurryinoc1, i1spiprs, 
                       c("ASV"))
nrow(i1spiprs_slurry)
all <- inner_join(i1spiprs_slurry, i1spips_i1rpips, 
                       c("ASV"))
nrow(all)
ABC <- inner_join(i1spips_i1rpips, i1spiprs, 
                       c("ASV"))
nrow(ABC)
ABD <- inner_join(i1spips_i1rpips, slurryinoc1, 
                       c("ASV"))
nrow(ABD)
BCD <- inner_join(i1spiprs_slurry, i1rpips, 
                       c("ASV"))
nrow(BCD)
```

i1spips = 308 = A i1rpips = 537 = B i1spiprs = 89 = C slurry = 20 = D

i1spips & i1rpips = 211 = A&B i1spips & i1spiprs = 63 = A&C i1spips &
slurry = 13 = A&D i1rpips & i1spiprs = 64 = B&C i1rpips & slurry = 12 =
B&D i1spiprs & slurry = 11 = C&D i1spips & i1rpips & i1spiprs & slurry =
10 = A&B&C&D i1spips & i1rpips & i1spiprs = = A&B&C = 58 i1spips &
i1rpips & slurry = = A&B&D = 12 i1rpips & i1spiprs & slurry = 10 = B&C&D

- Inoculum 1 graph
```{r inoculum 1 graph}
# use eulerr

inoc1 <-c("Shoot Post-Inoc Post-Stress" = 308, "Root Post-Inoc Post-Stress" = 537, "Shoot Post-Inoc Pre-Stress" = 89, "Slurry" = 20, 
                    "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress" = 211, "Shoot Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 63, "Shoot Post-Inoc Post-Stress&Slurry" = 13, "Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 64,"Root Post-Inoc Post-Stress&Slurry" = 12,
                    "Shoot Post-Inoc Pre-Stress&Slurry" = 11, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress&Slurry" = 10, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 58, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Slurry" = 12,
                    "Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress&Slurry" = 10)
fitinoc1 <- euler(inoc1,shape="ellipse")
plot(fitinoc1)

dev.copy(pdf, "figures/inoc1venn.pdf", height=8, width=8)
dev.off() 
```

##### Inoculum 4
```{r inoculum 4 loaded,results='hide'}
input <- load_taxa_table("data/ASV_table_postfiltering.txt", "data/map_loaded_strawberries.txt")

#inoc4
inputinoc4only = filter_data(input, 'Inoculum', keep_vals = '4')

#leaves post-inoc post stress
inputinoc4onlyshoot = filter_data(inputinoc4only, 'Type', keep_vals = 'Shoot')
inoc4shootpostinocpoststress = filter_data(inputinoc4onlyshoot, 'SampleType', keep_vals = 'post_inoc_post_stress')
i4spips<-as.data.frame(inoc4shootpostinocpoststress$data_loaded)
i4spips$ASV <-row.names(i4spips)

#roots post-inoc post-stress
inputinoc4onlyroot = filter_data(inputinoc4only, 'Type', keep_vals = 'Root')
inoc4rootpostinocpoststress = filter_data(inputinoc4onlyroot, 'SampleType', keep_vals = 'post_inoc_post_stress')
i4rpips<-as.data.frame(inoc4rootpostinocpoststress$data_loaded)
i4rpips$ASV <-row.names(i4rpips)

#leaves post-inoc pre-stress
inoc4shootpostinocprestress = filter_data(inputinoc4onlyshoot, 'SampleType', keep_vals = 'post_inoc_pre_stress')
i4spiprs<-as.data.frame(inoc4shootpostinocprestress$data_loaded)
i4spiprs$ASV <-row.names(i4spiprs)

#Then let's grab original inoculum. We will need to grab the original data frame again.
inputraw <- load_taxa_table("data/ASV_table_raw.txt", "data/strawberries_raw_full_data.txt")
slurryonly = filter_data(inputraw, 'SampleType', keep_vals = 'slurry')
slurryonlyinoc4 = filter_data(slurryonly, 'Type', keep_vals = 'Inoculum 4')
slurryinoc4<-as.data.frame(slurryonlyinoc4$data_loaded)
slurryinoc4$ASV <-row.names(slurryinoc4)

```

Let's count the number of rows and assess overlap:
```{r}
nrow(i4spips)
nrow(i4rpips)
nrow(i4spiprs)
nrow(slurryinoc4)

#### Let's see how these all overlap:

i4spips_i4rpips <- inner_join(i4spips, i4rpips, 
                       c("ASV"))
nrow(i4spips_i4rpips)
i4spips_i4spiprs <- inner_join(i4spips, i4spiprs, 
                       c("ASV"))
nrow(i4spips_i4spiprs)
i4spips_slurry <- inner_join(i4spips, slurryinoc4, 
                       c("ASV"))
nrow(i4spips_slurry)
i4rpips_i4spiprs <- inner_join(i4spiprs, i4rpips, 
                       c("ASV"))
nrow(i4rpips_i4spiprs)
i4rpips_slurry <- inner_join(slurryinoc4, i4rpips, 
                       c("ASV"))
nrow(i4rpips_slurry)
i4spiprs_slurry <- inner_join(slurryinoc4, i4spiprs, 
                       c("ASV"))
nrow(i4spiprs_slurry)
all <- inner_join(i4spiprs_slurry, i4spips_i4rpips, 
                       c("ASV"))
nrow(all)
ABC <- inner_join(i4spips_i4rpips, i4spiprs, 
                       c("ASV"))
nrow(ABC)
ABD <- inner_join(i4spips_i4rpips, slurryinoc4, 
                       c("ASV"))
nrow(ABD)
BCD <- inner_join(i4spiprs_slurry, i4rpips, 
                       c("ASV"))
nrow(BCD)
```

i4spips = 290 = A i4rpips = 519 = B i4spiprs = 108 = C slurry = 9 = D

i4spips & i4rpips = 187 = A&B i4spips & i4spiprs = 67 = A&C i4spips &
slurry = 5 = A&D i4rpips & i4spiprs = 71 = B&C i4rpips & slurry = 5 =
B&D i4spiprs & slurry = 2 = C&D i4spips & i4rpips & i4spiprs & slurry =
2 = A&B&C&D i4spips & i4rpips & i4spiprs = = A&B&C = 60 i4spips &
i4rpips & slurry = = A&B&D = 4 i4rpips & i4spiprs & slurry = 2 = B&C&D

- Inoculum 4 graph
```{r inoculum 4 graph}

inoc4 <-c("Shoot Post-Inoc Post-Stress" = 290, "Root Post-Inoc Post-Stress" = 519, "Shoot Post-Inoc Pre-Stress" = 108, "Slurry" = 9, 
                    "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress" = 187, "Shoot Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 67, "Shoot Post-Inoc Post-Stress&Slurry" = 5, "Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 71,"Root Post-Inoc Post-Stress&Slurry" = 5,
                    "Shoot Post-Inoc Pre-Stress&Slurry" = 2, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress&Slurry" = 2, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 60, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Slurry" = 4,
                    "Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress&Slurry" = 2)

fitinoc4 <- euler(inoc4,shape="ellipse")
plot(fitinoc4)

dev.copy(pdf, "figures/inoc4venn.pdf", height=8, width=8)
dev.off() 
```

##### Inoculum 5
```{r inoculum 5 loaded,results='hide'}
input <- load_taxa_table("data/ASV_table_postfiltering.txt", "data/map_loaded_strawberries.txt")

#inoc5
inputinoc5only = filter_data(input, 'Inoculum', keep_vals = '5')
#leaves post-inoc post stress
inputinoc5onlyshoot = filter_data(inputinoc5only, 'Type', keep_vals = 'Shoot')
inoc5shootpostinocpoststress = filter_data(inputinoc5onlyshoot, 'SampleType', keep_vals = 'post_inoc_post_stress')
i5spips<-as.data.frame(inoc5shootpostinocpoststress$data_loaded)
i5spips$ASV <-row.names(i5spips)

#roots post-inoc post-stress
inputinoc5onlyroot = filter_data(inputinoc5only, 'Type', keep_vals = 'Root')
inoc5rootpostinocpoststress = filter_data(inputinoc5onlyroot, 'SampleType', keep_vals = 'post_inoc_post_stress')

i5rpips<-as.data.frame(inoc5rootpostinocpoststress$data_loaded)
i5rpips$ASV <-row.names(i5rpips)

#leaves post-inoc pre-stress
inoc5shootpostinocprestress = filter_data(inputinoc5onlyshoot, 'SampleType', keep_vals = 'post_inoc_pre_stress')

i5spiprs<-as.data.frame(inoc5shootpostinocprestress$data_loaded)
i5spiprs$ASV <-row.names(i5spiprs)

#Then let's grab original inoculum. We will need to grab the original data frame again.
inputraw <- load_taxa_table("data/ASV_table_raw.txt", "data/strawberries_raw_full_data.txt")
slurryonly = filter_data(inputraw, 'SampleType', keep_vals = 'slurry')
slurryonlyinoc5 = filter_data(slurryonly, 'Type', keep_vals = 'Inoculum 5')
slurryinoc5<-as.data.frame(slurryonlyinoc5$data_loaded)
slurryinoc5$ASV <-row.names(slurryinoc5)

```

Let's count the number of rows and assess overlap
```{r}
nrow(i5spips)
nrow(i5rpips)
nrow(i5spiprs)
nrow(slurryinoc5)

##### Let's see how these all overlap:

i5spips_i5rpips <- inner_join(i5spips, i5rpips, 
                       c("ASV"))
nrow(i5spips_i5rpips)
i5spips_i5spiprs <- inner_join(i5spips, i5spiprs, 
                       c("ASV"))
nrow(i5spips_i5spiprs)
i5spips_slurry <- inner_join(i5spips, slurryinoc5, 
                       c("ASV"))
nrow(i5spips_slurry)
i5rpips_i5spiprs <- inner_join(i5spiprs, i5rpips, 
                       c("ASV"))
nrow(i5rpips_i5spiprs)
i5rpips_slurry <- inner_join(slurryinoc5, i5rpips, 
                       c("ASV"))
nrow(i5rpips_slurry)
i5spiprs_slurry <- inner_join(slurryinoc5, i5spiprs, 
                       c("ASV"))
nrow(i5spiprs_slurry)
all <- inner_join(i5spiprs_slurry, i5spips_i5rpips, 
                       c("ASV"))
nrow(all)
ABC <- inner_join(i5spips_i5rpips, i5spiprs, 
                       c("ASV"))
nrow(ABC)
ABD <- inner_join(i5spips_i5rpips, slurryinoc5, 
                       c("ASV"))
nrow(ABD)
BCD <- inner_join(i5spiprs_slurry, i5rpips, 
                       c("ASV"))
nrow(BCD)
```

i5spips = 174 = A i5rpips = 263 = B i5spiprs = 64 = C slurry = 50 = D

i5spips & i5rpips = 115 = A&B i5spips & i5spiprs = 43 = A&C i5spips &
slurry = 12 = A&D i5rpips & i5spiprs = 49 = B&C i5rpips & slurry = 12 =
B&D i5spiprs & slurry = 10 = C&D i5spips & i5rpips & i5spiprs & slurry =
10 = A&B&C&D i5spips & i5rpips & i5spiprs = A&B&C = 39 i5spips & i5rpips
& slurry = A&B&D = 12 i5rpips & i5spiprs & slurry = 10 = B&C&D

- Inoculum 5 graph
```{r inoculum 5 graph}

inoc5 <-c("Shoot Post-Inoc Post-Stress" = 174, "Root Post-Inoc Post-Stress" = 263, "Shoot Post-Inoc Pre-Stress" = 64, "Slurry" = 50, 
                    "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress" = 115, "Shoot Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 43, "Shoot Post-Inoc Post-Stress&Slurry" = 12, "Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 49,"Root Post-Inoc Post-Stress&Slurry" = 12,
                    "Shoot Post-Inoc Pre-Stress&Slurry" = 10, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress&Slurry" = 10, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress" = 39, "Shoot Post-Inoc Post-Stress&Root Post-Inoc Post-Stress&Slurry" = 12,
                    "Root Post-Inoc Post-Stress&Shoot Post-Inoc Pre-Stress&Slurry" = 10)
fitinoc5 <- euler(inoc5,shape="ellipse")
plot(fitinoc5)

dev.copy(pdf, "figures/inoc5venn.pdf", height=8, width=8)
dev.off() 
```
